// gsap@3.15.0/CSSPlugin.js
import he, { _isUndefined as Qe, _getSetter as ei, _plugins as ii, _checkPlugin as ni, _replaceRandom as Me, _colorExp as Ee, getUnit as G, _isString as We, _config as ie, PropTween as q, _parseRelative as me, _missingPlugin as ti, _sortPropTweensByPriority as si, _getCache as Ge, _getProperty as ri, _round as M, _ticker as Ae, GSCache as oi, _renderComplexString as fi, _colorStringFilter as li, _numWithUnitExp as de, _relExp as ai, _numExp as ci, _forEachName as ae } from "./gsap-core.js";
var Ye, U, te, be, ee, ke, Se, ui = function() {
  return typeof window < "u";
}, D = {}, Q = 180 / Math.PI, se = Math.PI / 180, ne = Math.atan2, Xe = 1e8, ye = /([A-Z])/g, hi = /(left|right|width|margin|padding|x)/i, _i = /[\s,\(]\S/, V = {
  autoAlpha: "opacity,visibility",
  scale: "scaleX,scaleY",
  alpha: "opacity"
}, ge = function(e, i) {
  return i.set(i.t, i.p, Math.round((i.s + i.c * e) * 1e4) / 1e4 + i.u, i);
}, di = function(e, i) {
  return i.set(i.t, i.p, e === 1 ? i.e : Math.round((i.s + i.c * e) * 1e4) / 1e4 + i.u, i);
}, mi = function(e, i) {
  return i.set(i.t, i.p, e ? Math.round((i.s + i.c * e) * 1e4) / 1e4 + i.u : i.b, i);
}, xi = function(e, i) {
  return i.set(i.t, i.p, e === 1 ? i.e : e ? Math.round((i.s + i.c * e) * 1e4) / 1e4 + i.u : i.b, i);
}, pi = function(e, i) {
  var n = i.s + i.c * e;
  i.set(i.t, i.p, ~~(n + (n < 0 ? -0.5 : 0.5)) + i.u, i);
}, Ue = function(e, i) {
  return i.set(i.t, i.p, e ? i.e : i.b, i);
}, Ie = function(e, i) {
  return i.set(i.t, i.p, e !== 1 ? i.b : i.e, i);
}, vi = function(e, i, n) {
  return e.style[i] = n;
}, gi = function(e, i, n) {
  return e.style.setProperty(i, n);
}, Oi = function(e, i, n) {
  return e._gsap[i] = n;
}, Pi = function(e, i, n) {
  return e._gsap.scaleX = e._gsap.scaleY = n;
}, wi = function(e, i, n, t, s) {
  var f = e._gsap;
  f.scaleX = f.scaleY = n, f.renderTransform(s, f);
}, bi = function(e, i, n, t, s) {
  var f = e._gsap;
  f[i] = n, f.renderTransform(s, f);
}, C = "transform", Y = C + "Origin", Si = function u(e, i) {
  var n = this, t = this.target, s = t.style, f = t._gsap;
  if (e in D && s) {
    if (this.tfm = this.tfm || {}, e !== "transform")
      e = V[e] || e, ~e.indexOf(",") ? e.split(",").forEach(function(o) {
        return n.tfm[o] = N(t, o);
      }) : this.tfm[e] = f.x ? f[e] : N(t, e), e === Y && (this.tfm.zOrigin = f.zOrigin);
    else
      return V.transform.split(",").forEach(function(o) {
        return u.call(n, o, i);
      });
    if (this.props.indexOf(C) >= 0)
      return;
    f.svg && (this.svgo = t.getAttribute("data-svg-origin"), this.props.push(Y, i, "")), e = C;
  }
  (s || i) && this.props.push(e, i, s[e]);
}, qe = function(e) {
  e.translate && (e.removeProperty("translate"), e.removeProperty("scale"), e.removeProperty("rotate"));
}, yi = function() {
  var e = this.props, i = this.target, n = i.style, t = i._gsap, s, f;
  for (s = 0; s < e.length; s += 3)
    e[s + 1] ? e[s + 1] === 2 ? i[e[s]](e[s + 2]) : i[e[s]] = e[s + 2] : e[s + 2] ? n[e[s]] = e[s + 2] : n.removeProperty(e[s].substr(0, 2) === "--" ? e[s] : e[s].replace(ye, "-$1").toLowerCase());
  if (this.tfm) {
    for (f in this.tfm)
      t[f] = this.tfm[f];
    t.svg && (t.renderTransform(), i.setAttribute("data-svg-origin", this.svgo || "")), s = Se(), (!s || !s.isStart) && !n[C] && (qe(n), t.zOrigin && n[Y] && (n[Y] += " " + t.zOrigin + "px", t.zOrigin = 0, t.renderTransform()), t.uncache = 1);
  }
}, Le = function(e, i) {
  var n = {
    target: e,
    props: [],
    revert: yi,
    save: Si
  };
  return e._gsap || he.core.getCache(e), i && e.style && e.nodeType && i.split(",").forEach(function(t) {
    return n.save(t);
  }), n;
}, je, Oe = function(e, i) {
  var n = U.createElementNS ? U.createElementNS((i || "http://www.w3.org/1999/xhtml").replace(/^https/, "http"), e) : U.createElement(e);
  return n && n.style ? n : U.createElement(e);
}, X = function u(e, i, n) {
  var t = getComputedStyle(e);
  return t[i] || t.getPropertyValue(i.replace(ye, "-$1").toLowerCase()) || t.getPropertyValue(i) || !n && u(e, re(i) || i, 1) || "";
}, ze = "O,Moz,ms,Ms,Webkit".split(","), re = function(e, i, n) {
  var t = i || ee, s = t.style, f = 5;
  if (e in s && !n)
    return e;
  for (e = e.charAt(0).toUpperCase() + e.substr(1); f-- && !(ze[f] + e in s); )
    ;
  return f < 0 ? null : (f === 3 ? "ms" : f >= 0 ? ze[f] : "") + e;
}, Pe = function() {
  ui() && window.document && (Ye = window, U = Ye.document, te = U.documentElement, ee = Oe("div") || {
    style: {}
  }, Oe("div"), C = re(C), Y = C + "Origin", ee.style.cssText = "border-width:0;line-height:0;position:absolute;padding:0", je = !!re("perspective"), Se = he.core.reverting, be = 1);
}, Fe = function(e) {
  var i = e.ownerSVGElement, n = Oe("svg", i && i.getAttribute("xmlns") || "http://www.w3.org/2000/svg"), t = e.cloneNode(!0), s;
  t.style.display = "block", n.appendChild(t), te.appendChild(n);
  try {
    s = t.getBBox();
  } catch {
  }
  return n.removeChild(t), te.removeChild(n), s;
}, Ve = function(e, i) {
  for (var n = i.length; n--; )
    if (e.hasAttribute(i[n]))
      return e.getAttribute(i[n]);
}, Ze = function(e) {
  var i, n;
  try {
    i = e.getBBox();
  } catch {
    i = Fe(e), n = 1;
  }
  return i && (i.width || i.height) || n || (i = Fe(e)), i && !i.width && !i.x && !i.y ? {
    x: +Ve(e, ["x", "cx", "x1"]) || 0,
    y: +Ve(e, ["y", "cy", "y1"]) || 0,
    width: 0,
    height: 0
  } : i;
}, He = function(e) {
  return !!(e.getCTM && (!e.parentNode || e.ownerSVGElement) && Ze(e));
}, L = function(e, i) {
  if (i) {
    var n = e.style, t;
    i in D && i !== Y && (i = C), n.removeProperty ? (t = i.substr(0, 2), (t === "ms" || i.substr(0, 6) === "webkit") && (i = "-" + i), n.removeProperty(t === "--" ? i : i.replace(ye, "-$1").toLowerCase())) : n.removeAttribute(i);
  }
}, I = function(e, i, n, t, s, f) {
  var o = new q(e._pt, i, n, 0, 1, f ? Ie : Ue);
  return e._pt = o, o.b = t, o.e = s, e._props.push(n), o;
}, Be = {
  deg: 1,
  rad: 1,
  turn: 1
}, Ci = {
  grid: 1,
  flex: 1
}, j = function u(e, i, n, t) {
  var s = parseFloat(n) || 0, f = (n + "").trim().substr((s + "").length) || "px", o = ee.style, c = hi.test(i), a = e.tagName.toLowerCase() === "svg", l = (a ? "client" : "offset") + (c ? "Width" : "Height"), h = 100, _ = t === "px", d = t === "%", m, r, g, O;
  if (t === f || !s || Be[t] || Be[f])
    return s;
  if (f !== "px" && !_ && (s = u(e, i, n, "px")), O = e.getCTM && He(e), (d || f === "%") && (D[i] || ~i.indexOf("adius")))
    return m = O ? e.getBBox()[c ? "width" : "height"] : e[l], M(d ? s / m * h : s / 100 * m);
  if (o[c ? "width" : "height"] = h + (_ ? f : t), r = t !== "rem" && ~i.indexOf("adius") || t === "em" && e.appendChild && !a ? e : e.parentNode, O && (r = (e.ownerSVGElement || {}).parentNode), (!r || r === U || !r.appendChild) && (r = U.body), g = r._gsap, g && d && g.width && c && g.time === Ae.time && !g.uncache)
    return M(s / g.width * h);
  if (d && (i === "height" || i === "width")) {
    var v = e.style[i];
    e.style[i] = h + t, m = e[l], v ? e.style[i] = v : L(e, i);
  } else
    (d || f === "%") && !Ci[X(r, "display")] && (o.position = X(e, "position")), r === e && (o.position = "static"), r.appendChild(ee), m = ee[l], r.removeChild(ee), o.position = "absolute";
  return c && d && (g = Ge(r), g.time = Ae.time, g.width = r[l]), M(_ ? m * s / h : m && s ? h / m * s : 0);
}, N = function(e, i, n, t) {
  var s;
  return be || Pe(), i in V && i !== "transform" && (i = V[i], ~i.indexOf(",") && (i = i.split(",")[0])), D[i] && i !== "transform" ? (s = ue(e, t), s = i !== "transformOrigin" ? s[i] : s.svg ? s.origin : pe(X(e, Y)) + " " + s.zOrigin + "px") : (s = e.style[i], (!s || s === "auto" || t || ~(s + "").indexOf("calc(")) && (s = xe[i] && xe[i](e, i, n) || X(e, i) || ri(e, i) || (i === "opacity" ? 1 : 0))), n && !~(s + "").trim().indexOf(" ") ? j(e, i, s, n) + n : s;
}, Ti = function(e, i, n, t) {
  if (!n || n === "none") {
    var s = re(i, e, 1), f = s && X(e, s, 1);
    f && f !== n ? (i = s, n = f) : i === "borderColor" && (n = X(e, "borderTopColor"));
  }
  var o = new q(this._pt, e.style, i, 0, 1, fi), c = 0, a = 0, l, h, _, d, m, r, g, O, v, w, P, x;
  if (o.b = n, o.e = t, n += "", t += "", t.substring(0, 6) === "var(--" && (t = X(e, t.substring(4, t.indexOf(")")))), t === "auto" && (r = e.style[i], e.style[i] = t, t = X(e, i) || t, r ? e.style[i] = r : L(e, i)), l = [n, t], li(l), n = l[0], t = l[1], _ = n.match(de) || [], x = t.match(de) || [], x.length) {
    for (; h = de.exec(t); )
      g = h[0], v = t.substring(c, h.index), m ? m = (m + 1) % 5 : (v.substr(-5) === "rgba(" || v.substr(-5) === "hsla(") && (m = 1), g !== (r = _[a++] || "") && (d = parseFloat(r) || 0, P = r.substr((d + "").length), g.charAt(1) === "=" && (g = me(d, g) + P), O = parseFloat(g), w = g.substr((O + "").length), c = de.lastIndex - w.length, w || (w = w || ie.units[i] || P, c === t.length && (t += w, o.e += w)), P !== w && (d = j(e, i, r, w) || 0), o._pt = {
        _next: o._pt,
        p: v || a === 1 ? v : ",",
        //note: SVG spec allows omission of comma/space when a negative sign is wedged between two numbers, like 2.5-5.3 instead of 2.5,-5.3 but when tweening, the negative value may switch to positive, so we insert the comma just in case.
        s: d,
        c: O - d,
        m: m && m < 4 || i === "zIndex" ? Math.round : 0
      });
    o.c = c < t.length ? t.substring(c, t.length) : "";
  } else
    o.r = i === "display" && t === "none" ? Ie : Ue;
  return ai.test(t) && (o.e = 0), this._pt = o, o;
}, Re = {
  top: "0%",
  bottom: "100%",
  left: "0%",
  right: "100%",
  center: "50%"
}, Mi = function(e) {
  var i = e.split(" "), n = i[0], t = i[1] || "50%";
  return (n === "top" || n === "bottom" || t === "left" || t === "right") && (e = n, n = t, t = e), i[0] = Re[n] || n, i[1] = Re[t] || t, i.join(" ");
}, Ei = function(e, i) {
  if (i.tween && i.tween._time === i.tween._dur) {
    var n = i.t, t = n.style, s = i.u, f = n._gsap, o, c, a;
    if (s === "all" || s === !0)
      t.cssText = "", c = 1;
    else
      for (s = s.split(","), a = s.length; --a > -1; )
        o = s[a], D[o] && (c = 1, o = o === "transformOrigin" ? Y : C), L(n, o);
    c && (L(n, C), f && (f.svg && n.removeAttribute("transform"), t.scale = t.rotate = t.translate = "none", ue(n, 1), f.uncache = 1, qe(t)));
  }
}, xe = {
  clearProps: function(e, i, n, t, s) {
    if (s.data !== "isFromStart") {
      var f = e._pt = new q(e._pt, i, n, 0, 0, Ei);
      return f.u = t, f.pr = -10, f.tween = s, e._props.push(n), 1;
    }
  }
  /* className feature (about 0.4kb gzipped).
  , className(plugin, target, property, endValue, tween) {
  	let _renderClassName = (ratio, data) => {
  			data.css.render(ratio, data.css);
  			if (!ratio || ratio === 1) {
  				let inline = data.rmv,
  					target = data.t,
  					p;
  				target.setAttribute("class", ratio ? data.e : data.b);
  				for (p in inline) {
  					_removeProperty(target, p);
  				}
  			}
  		},
  		_getAllStyles = (target) => {
  			let styles = {},
  				computed = getComputedStyle(target),
  				p;
  			for (p in computed) {
  				if (isNaN(p) && p !== "cssText" && p !== "length") {
  					styles[p] = computed[p];
  				}
  			}
  			_setDefaults(styles, _parseTransform(target, 1));
  			return styles;
  		},
  		startClassList = target.getAttribute("class"),
  		style = target.style,
  		cssText = style.cssText,
  		cache = target._gsap,
  		classPT = cache.classPT,
  		inlineToRemoveAtEnd = {},
  		data = {t:target, plugin:plugin, rmv:inlineToRemoveAtEnd, b:startClassList, e:(endValue.charAt(1) !== "=") ? endValue : startClassList.replace(new RegExp("(?:\\s|^)" + endValue.substr(2) + "(?![\\w-])"), "") + ((endValue.charAt(0) === "+") ? " " + endValue.substr(2) : "")},
  		changingVars = {},
  		startVars = _getAllStyles(target),
  		transformRelated = /(transform|perspective)/i,
  		endVars, p;
  	if (classPT) {
  		classPT.r(1, classPT.d);
  		_removeLinkedListItem(classPT.d.plugin, classPT, "_pt");
  	}
  	target.setAttribute("class", data.e);
  	endVars = _getAllStyles(target, true);
  	target.setAttribute("class", startClassList);
  	for (p in endVars) {
  		if (endVars[p] !== startVars[p] && !transformRelated.test(p)) {
  			changingVars[p] = endVars[p];
  			if (!style[p] && style[p] !== "0") {
  				inlineToRemoveAtEnd[p] = 1;
  			}
  		}
  	}
  	cache.classPT = plugin._pt = new PropTween(plugin._pt, target, "className", 0, 0, _renderClassName, data, 0, -11);
  	if (style.cssText !== cssText) { //only apply if things change. Otherwise, in cases like a background-image that's pulled dynamically, it could cause a refresh. See https://gsap.com/forums/topic/20368-possible-gsap-bug-switching-classnames-in-chrome/.
  		style.cssText = cssText; //we recorded cssText before we swapped classes and ran _getAllStyles() because in cases when a className tween is overwritten, we remove all the related tweening properties from that class change (otherwise class-specific stuff can't override properties we've directly set on the target's style object due to specificity).
  	}
  	_parseTransform(target, true); //to clear the caching of transforms
  	data.css = new gsap.plugins.css();
  	data.css.init(target, changingVars, tween);
  	plugin._props.push(...data.css._props);
  	return 1;
  }
  */
}, ce = [1, 0, 0, 1, 0, 0], $e = {}, Ke = function(e) {
  return e === "matrix(1, 0, 0, 1, 0, 0)" || e === "none" || !e;
}, Ne = function(e) {
  var i = X(e, C);
  return Ke(i) ? ce : i.substr(7).match(ci).map(M);
}, Ce = function(e, i) {
  var n = e._gsap || Ge(e), t = e.style, s = Ne(e), f, o, c, a;
  return n.svg && e.getAttribute("transform") ? (c = e.transform.baseVal.consolidate().matrix, s = [c.a, c.b, c.c, c.d, c.e, c.f], s.join(",") === "1,0,0,1,0,0" ? ce : s) : (s === ce && !e.offsetParent && e !== te && !n.svg && (c = t.display, t.display = "block", f = e.parentNode, (!f || !e.offsetParent && !e.getBoundingClientRect().width) && (a = 1, o = e.nextElementSibling, te.appendChild(e)), s = Ne(e), c ? t.display = c : L(e, "display"), a && (o ? f.insertBefore(e, o) : f ? f.appendChild(e) : te.removeChild(e))), i && s.length > 6 ? [s[0], s[1], s[4], s[5], s[12], s[13]] : s);
}, we = function(e, i, n, t, s, f) {
  var o = e._gsap, c = s || Ce(e, !0), a = o.xOrigin || 0, l = o.yOrigin || 0, h = o.xOffset || 0, _ = o.yOffset || 0, d = c[0], m = c[1], r = c[2], g = c[3], O = c[4], v = c[5], w = i.split(" "), P = parseFloat(w[0]) || 0, x = parseFloat(w[1]) || 0, y, b, S, p;
  n ? c !== ce && (b = d * g - m * r) && (S = P * (g / b) + x * (-r / b) + (r * v - g * O) / b, p = P * (-m / b) + x * (d / b) - (d * v - m * O) / b, P = S, x = p) : (y = Ze(e), P = y.x + (~w[0].indexOf("%") ? P / 100 * y.width : P), x = y.y + (~(w[1] || w[0]).indexOf("%") ? x / 100 * y.height : x)), t || t !== !1 && o.smooth ? (O = P - a, v = x - l, o.xOffset = h + (O * d + v * r) - O, o.yOffset = _ + (O * m + v * g) - v) : o.xOffset = o.yOffset = 0, o.xOrigin = P, o.yOrigin = x, o.smooth = !!t, o.origin = i, o.originIsAbsolute = !!n, e.style[Y] = "0px 0px", f && (I(f, o, "xOrigin", a, P), I(f, o, "yOrigin", l, x), I(f, o, "xOffset", h, o.xOffset), I(f, o, "yOffset", _, o.yOffset)), e.setAttribute("data-svg-origin", P + " " + x);
}, ue = function(e, i) {
  var n = e._gsap || new oi(e);
  if ("x" in n && !i && !n.uncache)
    return n;
  var t = e.style, s = n.scaleX < 0, f = "px", o = "deg", c = getComputedStyle(e), a = X(e, Y) || "0", l, h, _, d, m, r, g, O, v, w, P, x, y, b, S, p, T, E, A, z, Z, W, F, k, B, _e, oe, fe, H, Te, R, $;
  return l = h = _ = r = g = O = v = w = P = 0, d = m = 1, n.svg = !!(e.getCTM && He(e)), c.translate && ((c.translate !== "none" || c.scale !== "none" || c.rotate !== "none") && (t[C] = (c.translate !== "none" ? "translate3d(" + (c.translate + " 0 0").split(" ").slice(0, 3).join(", ") + ") " : "") + (c.rotate !== "none" ? "rotate(" + c.rotate + ") " : "") + (c.scale !== "none" ? "scale(" + c.scale.split(" ").join(",") + ") " : "") + (c[C] !== "none" ? c[C] : "")), t.scale = t.rotate = t.translate = "none"), b = Ce(e, n.svg), n.svg && (n.uncache ? (B = e.getBBox(), a = n.xOrigin - B.x + "px " + (n.yOrigin - B.y) + "px", k = "") : k = !i && e.getAttribute("data-svg-origin"), we(e, k || a, !!k || n.originIsAbsolute, n.smooth !== !1, b)), x = n.xOrigin || 0, y = n.yOrigin || 0, b !== ce && (E = b[0], A = b[1], z = b[2], Z = b[3], l = W = b[4], h = F = b[5], b.length === 6 ? (d = Math.sqrt(E * E + A * A), m = Math.sqrt(Z * Z + z * z), r = E || A ? ne(A, E) * Q : 0, v = z || Z ? ne(z, Z) * Q + r : 0, v && (m *= Math.abs(Math.cos(v * se))), n.svg && (l -= x - (x * E + y * z), h -= y - (x * A + y * Z))) : ($ = b[6], Te = b[7], oe = b[8], fe = b[9], H = b[10], R = b[11], l = b[12], h = b[13], _ = b[14], S = ne($, H), g = S * Q, S && (p = Math.cos(-S), T = Math.sin(-S), k = W * p + oe * T, B = F * p + fe * T, _e = $ * p + H * T, oe = W * -T + oe * p, fe = F * -T + fe * p, H = $ * -T + H * p, R = Te * -T + R * p, W = k, F = B, $ = _e), S = ne(-z, H), O = S * Q, S && (p = Math.cos(-S), T = Math.sin(-S), k = E * p - oe * T, B = A * p - fe * T, _e = z * p - H * T, R = Z * T + R * p, E = k, A = B, z = _e), S = ne(A, E), r = S * Q, S && (p = Math.cos(S), T = Math.sin(S), k = E * p + A * T, B = W * p + F * T, A = A * p - E * T, F = F * p - W * T, E = k, W = B), g && Math.abs(g) + Math.abs(r) > 359.9 && (g = r = 0, O = 180 - O), d = M(Math.sqrt(E * E + A * A + z * z)), m = M(Math.sqrt(F * F + $ * $)), S = ne(W, F), v = Math.abs(S) > 2e-4 ? S * Q : 0, P = R ? 1 / (R < 0 ? -R : R) : 0), n.svg && (k = e.getAttribute("transform"), n.forceCSS = e.setAttribute("transform", "") || !Ke(X(e, C)), k && e.setAttribute("transform", k))), Math.abs(v) > 90 && Math.abs(v) < 270 && (s ? (d *= -1, v += r <= 0 ? 180 : -180, r += r <= 0 ? 180 : -180) : (m *= -1, v += v <= 0 ? 180 : -180)), i = i || n.uncache, n.x = l - ((n.xPercent = l && (!i && n.xPercent || (Math.round(e.offsetWidth / 2) === Math.round(-l) ? -50 : 0))) ? e.offsetWidth * n.xPercent / 100 : 0) + f, n.y = h - ((n.yPercent = h && (!i && n.yPercent || (Math.round(e.offsetHeight / 2) === Math.round(-h) ? -50 : 0))) ? e.offsetHeight * n.yPercent / 100 : 0) + f, n.z = _ + f, n.scaleX = M(d), n.scaleY = M(m), n.rotation = M(r) + o, n.rotationX = M(g) + o, n.rotationY = M(O) + o, n.skewX = v + o, n.skewY = w + o, n.transformPerspective = P + f, (n.zOrigin = parseFloat(a.split(" ")[2]) || !i && n.zOrigin || 0) && (t[Y] = pe(a)), n.xOffset = n.yOffset = 0, n.force3D = ie.force3D, n.renderTransform = n.svg ? Yi : je ? Je : Ai, n.uncache = 0, n;
}, pe = function(e) {
  return (e = e.split(" "))[0] + " " + e[1];
}, ve = function(e, i, n) {
  var t = G(i);
  return M(parseFloat(i) + parseFloat(j(e, "x", n + "px", t))) + t;
}, Ai = function(e, i) {
  i.z = "0px", i.rotationY = i.rotationX = "0deg", i.force3D = 0, Je(e, i);
}, K = "0deg", le = "0px", J = ") ", Je = function(e, i) {
  var n = i || this, t = n.xPercent, s = n.yPercent, f = n.x, o = n.y, c = n.z, a = n.rotation, l = n.rotationY, h = n.rotationX, _ = n.skewX, d = n.skewY, m = n.scaleX, r = n.scaleY, g = n.transformPerspective, O = n.force3D, v = n.target, w = n.zOrigin, P = "", x = O === "auto" && e && e !== 1 || O === !0;
  if (w && (h !== K || l !== K)) {
    var y = parseFloat(l) * se, b = Math.sin(y), S = Math.cos(y), p;
    y = parseFloat(h) * se, p = Math.cos(y), f = ve(v, f, b * p * -w), o = ve(v, o, -Math.sin(y) * -w), c = ve(v, c, S * p * -w + w);
  }
  g !== le && (P += "perspective(" + g + J), (t || s) && (P += "translate(" + t + "%, " + s + "%) "), (x || f !== le || o !== le || c !== le) && (P += c !== le || x ? "translate3d(" + f + ", " + o + ", " + c + ") " : "translate(" + f + ", " + o + J), a !== K && (P += "rotate(" + a + J), l !== K && (P += "rotateY(" + l + J), h !== K && (P += "rotateX(" + h + J), (_ !== K || d !== K) && (P += "skew(" + _ + ", " + d + J), (m !== 1 || r !== 1) && (P += "scale(" + m + ", " + r + J), v.style[C] = P || "translate(0, 0)";
}, Yi = function(e, i) {
  var n = i || this, t = n.xPercent, s = n.yPercent, f = n.x, o = n.y, c = n.rotation, a = n.skewX, l = n.skewY, h = n.scaleX, _ = n.scaleY, d = n.target, m = n.xOrigin, r = n.yOrigin, g = n.xOffset, O = n.yOffset, v = n.forceCSS, w = parseFloat(f), P = parseFloat(o), x, y, b, S, p;
  c = parseFloat(c), a = parseFloat(a), l = parseFloat(l), l && (l = parseFloat(l), a += l, c += l), c || a ? (c *= se, a *= se, x = Math.cos(c) * h, y = Math.sin(c) * h, b = Math.sin(c - a) * -_, S = Math.cos(c - a) * _, a && (l *= se, p = Math.tan(a - l), p = Math.sqrt(1 + p * p), b *= p, S *= p, l && (p = Math.tan(l), p = Math.sqrt(1 + p * p), x *= p, y *= p)), x = M(x), y = M(y), b = M(b), S = M(S)) : (x = h, S = _, y = b = 0), (w && !~(f + "").indexOf("px") || P && !~(o + "").indexOf("px")) && (w = j(d, "x", f, "px"), P = j(d, "y", o, "px")), (m || r || g || O) && (w = M(w + m - (m * x + r * b) + g), P = M(P + r - (m * y + r * S) + O)), (t || s) && (p = d.getBBox(), w = M(w + t / 100 * p.width), P = M(P + s / 100 * p.height)), p = "matrix(" + x + "," + y + "," + b + "," + S + "," + w + "," + P + ")", d.setAttribute("transform", p), v && (d.style[C] = p);
}, ki = function(e, i, n, t, s) {
  var f = 360, o = We(s), c = parseFloat(s) * (o && ~s.indexOf("rad") ? Q : 1), a = c - t, l = t + a + "deg", h, _;
  return o && (h = s.split("_")[1], h === "short" && (a %= f, a !== a % (f / 2) && (a += a < 0 ? f : -f)), h === "cw" && a < 0 ? a = (a + f * Xe) % f - ~~(a / f) * f : h === "ccw" && a > 0 && (a = (a - f * Xe) % f - ~~(a / f) * f)), e._pt = _ = new q(e._pt, i, n, t, a, di), _.e = l, _.u = "deg", e._props.push(n), _;
}, De = function(e, i) {
  for (var n in i)
    e[n] = i[n];
  return e;
}, Xi = function(e, i, n) {
  var t = De({}, n._gsap), s = "perspective,force3D,transformOrigin,svgOrigin", f = n.style, o, c, a, l, h, _, d, m;
  t.svg ? (a = n.getAttribute("transform"), n.setAttribute("transform", ""), f[C] = i, o = ue(n, 1), L(n, C), n.setAttribute("transform", a)) : (a = getComputedStyle(n)[C], f[C] = i, o = ue(n, 1), f[C] = a);
  for (c in D)
    a = t[c], l = o[c], a !== l && s.indexOf(c) < 0 && (d = G(a), m = G(l), h = d !== m ? j(n, c, a, m) : parseFloat(a), _ = parseFloat(l), e._pt = new q(e._pt, o, c, h, _ - h, ge), e._pt.u = m || 0, e._props.push(c));
  De(o, t);
};
ae("padding,margin,Width,Radius", function(u, e) {
  var i = "Top", n = "Right", t = "Bottom", s = "Left", f = (e < 3 ? [i, n, t, s] : [i + s, i + n, t + n, t + s]).map(function(o) {
    return e < 2 ? u + o : "border" + o + u;
  });
  xe[e > 1 ? "border" + u : u] = function(o, c, a, l, h) {
    var _, d;
    if (arguments.length < 4)
      return _ = f.map(function(m) {
        return N(o, m, a);
      }), d = _.join(" "), d.split(_[0]).length === 5 ? _[0] : d;
    _ = (l + "").split(" "), d = {}, f.forEach(function(m, r) {
      return d[m] = _[r] = _[r] || _[(r - 1) / 2 | 0];
    }), o.init(c, d, h);
  };
});
var zi = {
  name: "css",
  register: Pe,
  targetTest: function(e) {
    return e.style && e.nodeType;
  },
  init: function(e, i, n, t, s) {
    var f = this._props, o = e.style, c = n.vars.startAt, a, l, h, _, d, m, r, g, O, v, w, P, x, y, b, S, p;
    be || Pe(), this.styles = this.styles || Le(e), S = this.styles.props, this.tween = n;
    for (r in i)
      if (r !== "autoRound" && (l = i[r], !(ii[r] && ni(r, i, n, t, e, s)))) {
        if (d = typeof l, m = xe[r], d === "function" && (l = l.call(n, t, e, s), d = typeof l), d === "string" && ~l.indexOf("random(") && (l = Me(l)), m)
          m(this, e, r, l, n) && (b = 1);
        else if (r.substr(0, 2) === "--")
          a = (getComputedStyle(e).getPropertyValue(r) + "").trim(), l += "", Ee.lastIndex = 0, Ee.test(a) || (g = G(a), O = G(l), O ? g !== O && (a = j(e, r, a, O) + O) : g && (l += g)), this.add(o, "setProperty", a, l, t, s, 0, 0, r), f.push(r), S.push(r, 0, o[r]);
        else if (d !== "undefined") {
          if (c && r in c ? (a = typeof c[r] == "function" ? c[r].call(n, t, e, s) : c[r], We(a) && ~a.indexOf("random(") && (a = Me(a)), G(a + "") || a === "auto" || (a += ie.units[r] || G(N(e, r)) || ""), (a + "").charAt(1) === "=" && (a = N(e, r))) : a = N(e, r), _ = parseFloat(a), v = d === "string" && l.charAt(1) === "=" && l.substr(0, 2), v && (l = l.substr(2)), h = parseFloat(l), r in V && (r === "autoAlpha" && (_ === 1 && N(e, "visibility") === "hidden" && h && (_ = 0), S.push("visibility", 0, o.visibility), I(this, o, "visibility", _ ? "inherit" : "hidden", h ? "inherit" : "hidden", !h)), r !== "scale" && r !== "transform" && (r = V[r], ~r.indexOf(",") && (r = r.split(",")[0]))), w = r in D, w) {
            if (this.styles.save(r), p = l, d === "string" && l.substring(0, 6) === "var(--") {
              if (l = X(e, l.substring(4, l.indexOf(")"))), l.substring(0, 5) === "calc(") {
                var T = e.style.perspective;
                e.style.perspective = l, l = X(e, "perspective"), T ? e.style.perspective = T : L(e, "perspective");
              }
              h = parseFloat(l);
            }
            if (P || (x = e._gsap, x.renderTransform && !i.parseTransform || ue(e, i.parseTransform), y = i.smoothOrigin !== !1 && x.smooth, P = this._pt = new q(this._pt, o, C, 0, 1, x.renderTransform, x, 0, -1), P.dep = 1), r === "scale")
              this._pt = new q(this._pt, x, "scaleY", x.scaleY, (v ? me(x.scaleY, v + h) : h) - x.scaleY || 0, ge), this._pt.u = 0, f.push("scaleY", r), r += "X";
            else if (r === "transformOrigin") {
              S.push(Y, 0, o[Y]), l = Mi(l), x.svg ? we(e, l, 0, y, 0, this) : (O = parseFloat(l.split(" ")[2]) || 0, O !== x.zOrigin && I(this, x, "zOrigin", x.zOrigin, O), I(this, o, r, pe(a), pe(l)));
              continue;
            } else if (r === "svgOrigin") {
              we(e, l, 1, y, 0, this);
              continue;
            } else if (r in $e) {
              ki(this, x, r, _, v ? me(_, v + l) : l);
              continue;
            } else if (r === "smoothOrigin") {
              I(this, x, "smooth", x.smooth, l);
              continue;
            } else if (r === "force3D") {
              x[r] = l;
              continue;
            } else if (r === "transform") {
              Xi(this, l, e);
              continue;
            }
          } else r in o || (r = re(r) || r);
          if (w || (h || h === 0) && (_ || _ === 0) && !_i.test(l) && r in o)
            g = (a + "").substr((_ + "").length), h || (h = 0), O = G(l) || (r in ie.units ? ie.units[r] : g), g !== O && (_ = j(e, r, a, O)), this._pt = new q(this._pt, w ? x : o, r, _, (v ? me(_, v + h) : h) - _, !w && (O === "px" || r === "zIndex") && i.autoRound !== !1 ? pi : ge), this._pt.u = O || 0, w && p !== l ? (this._pt.b = a, this._pt.e = p, this._pt.r = xi) : g !== O && O !== "%" && (this._pt.b = a, this._pt.r = mi);
          else if (r in o)
            Ti.call(this, e, r, a, v ? v + l : l);
          else if (r in e)
            this.add(e, r, a || e[r], v ? v + l : l, t, s);
          else if (r !== "parseTransform") {
            ti(r, l);
            continue;
          }
          w || (r in o ? S.push(r, 0, o[r]) : typeof e[r] == "function" ? S.push(r, 2, e[r]()) : S.push(r, 1, a || e[r])), f.push(r);
        }
      }
    b && si(this);
  },
  render: function(e, i) {
    if (i.tween._time || !Se())
      for (var n = i._pt; n; )
        n.r(e, n.d), n = n._next;
    else
      i.styles.revert();
  },
  get: N,
  aliases: V,
  getSetter: function(e, i, n) {
    var t = V[i];
    return t && t.indexOf(",") < 0 && (i = t), i in D && i !== Y && (e._gsap.x || N(e, "x")) ? n && ke === n ? i === "scale" ? Pi : Oi : (ke = n || {}) && (i === "scale" ? wi : bi) : e.style && !Qe(e.style[i]) ? vi : ~i.indexOf("-") ? gi : ei(e, i);
  },
  core: {
    _removeProperty: L,
    _getMatrix: Ce
  }
};
he.utils.checkPrefix = re;
he.core.getStyleSaver = Le;
(function(u, e, i, n) {
  var t = ae(u + "," + e + "," + i, function(s) {
    D[s] = 1;
  });
  ae(e, function(s) {
    ie.units[s] = "deg", $e[s] = 1;
  }), V[t[13]] = u + "," + e, ae(n, function(s) {
    var f = s.split(":");
    V[f[1]] = t[f[0]];
  });
})("x,y,z,scale,scaleX,scaleY,xPercent,yPercent", "rotation,rotationX,rotationY,skewX,skewY", "transform,transformOrigin,svgOrigin,force3D,smoothOrigin,transformPerspective", "0:translateX,1:translateY,2:translateZ,8:rotate,8:rotationZ,8:rotateZ,9:rotateX,10:rotateY");
ae("x,y,z,top,right,bottom,left,width,height,fontSize,padding,margin,perspective", function(u) {
  ie.units[u] = "px";
});
he.registerPlugin(zi);
export {
  zi as CSSPlugin,
  Oe as _createElement,
  Ze as _getBBox,
  re as checkPrefix,
  zi as default
};
