// gsap@3.15.0/gsap-core.js
function tt(u) {
  if (u === void 0)
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  return u;
}
function Ee(u, t) {
  u.prototype = Object.create(t.prototype), u.prototype.constructor = u, u.__proto__ = t;
}
var Tt = {
  autoSleep: 120,
  force3D: "auto",
  nullTargetWarn: 1,
  units: {
    lineHeight: ""
  }
}, At = {
  duration: 0.5,
  overwrite: !1,
  delay: 0
}, le, F, S, K = 1e8, C = 1 / K, re = Math.PI * 2, gr = re / 4, mr = 0, ze = Math.sqrt, yr = Math.cos, Tr = Math.sin, I = function(t) {
  return typeof t == "string";
}, O = function(t) {
  return typeof t == "function";
}, et = function(t) {
  return typeof t == "number";
}, Fe = function(t) {
  return typeof t > "u";
}, j = function(t) {
  return typeof t == "object";
}, q = function(t) {
  return t !== !1;
}, de = function() {
  return typeof window < "u";
}, It = function(t) {
  return O(t) || I(t);
}, Le = typeof ArrayBuffer == "function" && ArrayBuffer.isView || function() {
}, L = Array.isArray, vr = /random\([^)]+\)/g, xr = /,\s*/g, De = /(?:-?\.?\d|\.)+/gi, kr = /[-+=.]*\d+[.e\-+]*\d*[e\-+]*\d*/g, Ie = /[-+=.]*\d+[.e-]*\d*[a-z%]*/g, Zt = /[-+=.]*\d+\.?\d*(?:e-|e\+)?\d*/gi, br = /[+-]=-?[.\d]+/, wr = /[^,'"\[\]\s]+/gi, Dr = /^[+\-=e\s\d]*\d+[.\d]*([a-z]*|%)\s*$/i, A, $, ie, ce, V = {}, Bt = {}, Ne, Ue = function(t) {
  return (Bt = vt(t, V)) && ct;
}, qe = function(t, e) {
  return console.warn("Invalid property", t, "set to", e, "Missing plugin? gsap.registerPlugin()");
}, Ot = function(t, e) {
  return !e && console.warn(t);
}, Be = function(t, e) {
  return t && (V[t] = e) && Bt && (Bt[t] = e) || V;
}, Et = function() {
  return 0;
}, Rr = {
  suppressEvents: !0,
  isStart: !0,
  kill: !1
}, Nt = {
  suppressEvents: !0,
  kill: !1
}, Pr = {
  suppressEvents: !0
}, pe = {}, st = [], ne = {}, Ve, Y = {}, $t = {}, Re = 30, Ut = [], ge = "", me = function(t) {
  var e = t[0], i, r;
  if (j(e) || O(e) || (t = [t]), !(i = (e._gsap || {}).harness)) {
    for (r = Ut.length; r-- && !Ut[r].targetTest(e); )
      ;
    i = Ut[r];
  }
  for (r = t.length; r--; )
    t[r] && (t[r]._gsap || (t[r]._gsap = new ei(t[r], i))) || t.splice(r, 1);
  return t;
}, Pt = function(t) {
  return t._gsap || me(X(t))[0]._gsap;
}, Cr = function(t, e, i) {
  return (i = t[e]) && O(i) ? t[e]() : Fe(i) && t.getAttribute && t.getAttribute(e) || i;
}, rt = function(t, e) {
  return (t = t.split(",")).forEach(e) || t;
}, Sr = function(t) {
  return Math.round(t * 1e5) / 1e5 || 0;
}, M = function(t) {
  return Math.round(t * 1e7) / 1e7 || 0;
}, Ge = function(t, e) {
  var i = e.charAt(0), r = parseFloat(e.substr(2));
  return t = parseFloat(t), i === "+" ? t + r : i === "-" ? t - r : i === "*" ? t * r : t / r;
}, Mr = function(t, e) {
  for (var i = e.length, r = 0; t.indexOf(e[r]) < 0 && ++r < i; )
    ;
  return r < i;
}, Vt = function() {
  var t = st.length, e = st.slice(0), i, r;
  for (ne = {}, st.length = 0, i = 0; i < t; i++)
    r = e[i], r && r._lazy && (r.render(r._lazy[0], r._lazy[1], !0)._lazy = 0);
}, ye = function(t) {
  return !!(t._initted || t._startAt || t.add);
}, We = function(t, e, i, r) {
  st.length && !F && Vt(), t.render(e, i, !!(F && e < 0 && ye(t))), st.length && !F && Vt();
}, Qe = function(t) {
  var e = parseFloat(t);
  return (e || e === 0) && (t + "").match(wr).length < 2 ? e : I(t) ? t.trim() : t;
}, Ye = function(t) {
  return t;
}, G = function(t, e) {
  for (var i in e)
    i in t || (t[i] = e[i]);
  return t;
}, Ar = function(t) {
  return function(e, i) {
    for (var r in i)
      r in e || r === "duration" && t || r === "ease" || (e[r] = i[r]);
  };
}, vt = function(t, e) {
  for (var i in e)
    t[i] = e[i];
  return t;
}, Pe = function u(t, e) {
  for (var i in e)
    i !== "__proto__" && i !== "constructor" && i !== "prototype" && (t[i] = j(e[i]) ? u(t[i] || (t[i] = {}), e[i]) : e[i]);
  return t;
}, Gt = function(t, e) {
  var i = {}, r;
  for (r in t)
    r in e || (i[r] = t[r]);
  return i;
}, Ct = function(t) {
  var e = t.parent || A, i = t.keyframes ? Ar(L(t.keyframes)) : G;
  if (q(t.inherit))
    for (; e; )
      i(t, e.vars.defaults), e = e.parent || e._dp;
  return t;
}, Or = function(t, e) {
  for (var i = t.length, r = i === e.length; r && i-- && t[i] === e[i]; )
    ;
  return i < 0;
}, Ke = function(t, e, i, r, n) {
  var s = t[r], a;
  if (n)
    for (a = e[n]; s && s[n] > a; )
      s = s._prev;
  return s ? (e._next = s._next, s._next = e) : (e._next = t[i], t[i] = e), e._next ? e._next._prev = e : t[r] = e, e._prev = s, e.parent = e._dp = t, e;
}, Yt = function(t, e, i, r) {
  i === void 0 && (i = "_first"), r === void 0 && (r = "_last");
  var n = e._prev, s = e._next;
  n ? n._next = s : t[i] === e && (t[i] = s), s ? s._prev = n : t[r] === e && (t[r] = n), e._next = e._prev = e.parent = null;
}, at = function(t, e) {
  t.parent && (!e || t.parent.autoRemoveChildren) && t.parent.remove && t.parent.remove(t), t._act = 0;
}, ft = function(t, e) {
  if (t && (!e || e._end > t._dur || e._start < 0))
    for (var i = t; i; )
      i._dirty = 1, i = i.parent;
  return t;
}, Er = function(t) {
  for (var e = t.parent; e && e.parent; )
    e._dirty = 1, e.totalDuration(), e = e.parent;
  return t;
}, se = function(t, e, i, r) {
  return t._startAt && (F ? t._startAt.revert(Nt) : t.vars.immediateRender && !t.vars.autoRevert || t._startAt.render(e, !0, r));
}, zr = function u(t) {
  return !t || t._ts && u(t.parent);
}, Ce = function(t) {
  return t._repeat ? xt(t._tTime, t = t.duration() + t._rDelay) * t : 0;
}, xt = function(t, e) {
  var i = Math.floor(t = M(t / e));
  return t && i === t ? i - 1 : i;
}, Wt = function(t, e) {
  return (t - e._start) * e._ts + (e._ts >= 0 ? 0 : e._dirty ? e.totalDuration() : e._tDur);
}, Kt = function(t) {
  return t._end = M(t._start + (t._tDur / Math.abs(t._ts || t._rts || C) || 0));
}, Xt = function(t, e) {
  var i = t._dp;
  return i && i.smoothChildTiming && t._ts && (t._start = M(i._time - (t._ts > 0 ? e / t._ts : ((t._dirty ? t.totalDuration() : t._tDur) - e) / -t._ts)), Kt(t), i._dirty || ft(i, t)), t;
}, Xe = function(t, e) {
  var i;
  if ((e._time || !e._dur && e._initted || e._start < t._time && (e._dur || !e.add)) && (i = Wt(t.rawTime(), e), (!e._dur || Lt(0, e.totalDuration(), i) - e._tTime > C) && e.render(i, !0)), ft(t, e)._dp && t._initted && t._time >= t._dur && t._ts) {
    if (t._dur < t.duration())
      for (i = t; i._dp; )
        i.rawTime() >= 0 && i.totalTime(i._tTime), i = i._dp;
    t._zTime = -C;
  }
}, H = function(t, e, i, r) {
  return e.parent && at(e), e._start = M((et(i) ? i : i || t !== A ? Q(t, i, e) : t._time) + e._delay), e._end = M(e._start + (e.totalDuration() / Math.abs(e.timeScale()) || 0)), Ke(t, e, "_first", "_last", t._sort ? "_start" : 0), ae(e) || (t._recent = e), r || Xe(t, e), t._ts < 0 && Xt(t, t._tTime), t;
}, Ze = function(t, e) {
  return (V.ScrollTrigger || qe("scrollTrigger", e)) && V.ScrollTrigger.create(e, t);
}, $e = function(t, e, i, r, n) {
  if (xe(t, e, n), !t._initted)
    return 1;
  if (!i && t._pt && !F && (t._dur && t.vars.lazy !== !1 || !t._dur && t.vars.lazy) && Ve !== Z.frame)
    return st.push(t), t._lazy = [n, r], 1;
}, Fr = function u(t) {
  var e = t.parent;
  return e && e._ts && e._initted && !e._lock && (e.rawTime() < 0 || u(e));
}, ae = function(t) {
  var e = t.data;
  return e === "isFromStart" || e === "isStart";
}, Lr = function(t, e, i, r) {
  var n = t.ratio, s = e < 0 || !e && (!t._start && Fr(t) && !(!t._initted && ae(t)) || (t._ts < 0 || t._dp._ts < 0) && !ae(t)) ? 0 : 1, a = t._rDelay, o = 0, _, h, f;
  if (a && t._repeat && (o = Lt(0, t._tDur, e), h = xt(o, a), t._yoyo && h & 1 && (s = 1 - s), h !== xt(t._tTime, a) && (n = 1 - s, t.vars.repeatRefresh && t._initted && t.invalidate())), s !== n || F || r || t._zTime === C || !e && t._zTime) {
    if (!t._initted && $e(t, e, r, i, o))
      return;
    for (f = t._zTime, t._zTime = e || (i ? C : 0), i || (i = e && !f), t.ratio = s, t._from && (s = 1 - s), t._time = 0, t._tTime = o, _ = t._pt; _; )
      _.r(s, _.d), _ = _._next;
    e < 0 && se(t, e, i, !0), t._onUpdate && !i && B(t, "onUpdate"), o && t._repeat && !i && t.parent && B(t, "onRepeat"), (e >= t._tDur || e < 0) && t.ratio === s && (s && at(t, 1), !i && !F && (B(t, s ? "onComplete" : "onReverseComplete", !0), t._prom && t._prom()));
  } else t._zTime || (t._zTime = e);
}, Ir = function(t, e, i) {
  var r;
  if (i > e)
    for (r = t._first; r && r._start <= i; ) {
      if (r.data === "isPause" && r._start > e)
        return r;
      r = r._next;
    }
  else
    for (r = t._last; r && r._start >= i; ) {
      if (r.data === "isPause" && r._start < e)
        return r;
      r = r._prev;
    }
}, kt = function(t, e, i, r) {
  var n = t._repeat, s = M(e) || 0, a = t._tTime / t._tDur;
  return a && !r && (t._time *= s / t._dur), t._dur = s, t._tDur = n ? n < 0 ? 1e10 : M(s * (n + 1) + t._rDelay * n) : s, a > 0 && !r && Xt(t, t._tTime = t._tDur * a), t.parent && Kt(t), i || ft(t.parent, t), t;
}, Se = function(t) {
  return t instanceof U ? ft(t) : kt(t, t._dur);
}, Nr = {
  _start: 0,
  endTime: Et,
  totalDuration: Et
}, Q = function u(t, e, i) {
  var r = t.labels, n = t._recent || Nr, s = t.duration() >= K ? n.endTime(!1) : t._dur, a, o, _;
  return I(e) && (isNaN(e) || e in r) ? (o = e.charAt(0), _ = e.substr(-1) === "%", a = e.indexOf("="), o === "<" || o === ">" ? (a >= 0 && (e = e.replace(/=/, "")), (o === "<" ? n._start : n.endTime(n._repeat >= 0)) + (parseFloat(e.substr(1)) || 0) * (_ ? (a < 0 ? n : i).totalDuration() / 100 : 1)) : a < 0 ? (e in r || (r[e] = s), r[e]) : (o = parseFloat(e.charAt(a - 1) + e.substr(a + 1)), _ && i && (o = o / 100 * (L(i) ? i[0] : i).totalDuration()), a > 1 ? u(t, e.substr(0, a - 1), i) + o : s + o)) : e == null ? s : +e;
}, St = function(t, e, i) {
  var r = et(e[1]), n = (r ? 2 : 1) + (t < 2 ? 0 : 1), s = e[n], a, o;
  if (r && (s.duration = e[1]), s.parent = i, t) {
    for (a = s, o = i; o && !("immediateRender" in a); )
      a = o.vars.defaults || {}, o = q(o.vars.inherit) && o.parent;
    s.immediateRender = q(a.immediateRender), t < 2 ? s.runBackwards = 1 : s.startAt = e[n - 1];
  }
  return new z(e[0], s, e[n + 1]);
}, ot = function(t, e) {
  return t || t === 0 ? e(t) : e;
}, Lt = function(t, e, i) {
  return i < t ? t : i > e ? e : i;
}, ut = function(t, e) {
  return !I(t) || !(e = Dr.exec(t)) ? "" : e[1];
}, Ur = function(t, e, i) {
  return ot(i, function(r) {
    return Lt(t, e, r);
  });
}, ue = [].slice, He = function(t, e) {
  return t && j(t) && "length" in t && (!e && !t.length || t.length - 1 in t && j(t[0])) && !t.nodeType && t !== $;
}, qr = function(t, e, i) {
  return i === void 0 && (i = []), t.forEach(function(r) {
    var n;
    return I(r) && !e || He(r, 1) ? (n = i).push.apply(n, X(r)) : i.push(r);
  }) || i;
}, X = function(t, e, i) {
  return S && !e && S.selector ? S.selector(t) : I(t) && !i && (ie || !bt()) ? ue.call((e || ce).querySelectorAll(t), 0) : L(t) ? qr(t, i) : He(t) ? ue.call(t, 0) : t ? [t] : [];
}, oe = function(t) {
  return t = X(t)[0] || Ot("Invalid scope") || {}, function(e) {
    var i = t.current || t.nativeElement || t;
    return X(e, i.querySelectorAll ? i : i === t ? Ot("Invalid scope") || ce.createElement("div") : t);
  };
}, je = function(t) {
  return t.sort(function() {
    return 0.5 - Math.random();
  });
}, Je = function(t) {
  if (O(t))
    return t;
  var e = j(t) ? t : {
    each: t
  }, i = lt(e.ease), r = e.from || 0, n = parseFloat(e.base) || 0, s = {}, a = r > 0 && r < 1, o = isNaN(r) || a, _ = e.axis, h = r, f = r;
  return I(r) ? h = f = {
    center: 0.5,
    edges: 0.5,
    end: 1
  }[r] || 0 : !a && o && (h = r[0], f = r[1]), function(l, d, g) {
    var c = (g || e).length, p = s[c], m, y, v, x, T, D, k, R, w;
    if (!p) {
      if (w = e.grid === "auto" ? 0 : (e.grid || [1, K])[1], !w) {
        for (k = -K; k < (k = g[w++].getBoundingClientRect().left) && w < c; )
          ;
        w < c && w--;
      }
      for (p = s[c] = [], m = o ? Math.min(w, c) * h - 0.5 : r % w, y = w === K ? 0 : o ? c * f / w - 0.5 : r / w | 0, k = 0, R = K, D = 0; D < c; D++)
        v = D % w - m, x = y - (D / w | 0), p[D] = T = _ ? Math.abs(_ === "y" ? x : v) : ze(v * v + x * x), T > k && (k = T), T < R && (R = T);
      r === "random" && je(p), p.max = k - R, p.min = R, p.v = c = (parseFloat(e.amount) || parseFloat(e.each) * (w > c ? c - 1 : _ ? _ === "y" ? c / w : w : Math.max(w, c / w)) || 0) * (r === "edges" ? -1 : 1), p.b = c < 0 ? n - c : n, p.u = ut(e.amount || e.each) || 0, i = i && c < 0 ? ti(i) : i;
    }
    return c = (p[l] - p.min) / p.max || 0, M(p.b + (i ? i(c) : c) * p.v) + p.u;
  };
}, _e = function(t) {
  var e = Math.pow(10, ((t + "").split(".")[1] || "").length);
  return function(i) {
    var r = M(Math.round(parseFloat(i) / t) * t * e);
    return (r - r % 1) / e + (et(i) ? 0 : ut(i));
  };
}, tr = function(t, e) {
  var i = L(t), r, n;
  return !i && j(t) && (r = i = t.radius || K, t.values ? (t = X(t.values), (n = !et(t[0])) && (r *= r)) : t = _e(t.increment)), ot(e, i ? O(t) ? function(s) {
    return n = t(s), Math.abs(n - s) <= r ? n : s;
  } : function(s) {
    for (var a = parseFloat(n ? s.x : s), o = parseFloat(n ? s.y : 0), _ = K, h = 0, f = t.length, l, d; f--; )
      n ? (l = t[f].x - a, d = t[f].y - o, l = l * l + d * d) : l = Math.abs(t[f] - a), l < _ && (_ = l, h = f);
    return h = !r || _ <= r ? t[h] : s, n || h === s || et(s) ? h : h + ut(s);
  } : _e(t));
}, er = function(t, e, i, r) {
  return ot(L(t) ? !e : i === !0 ? !!(i = 0) : !r, function() {
    return L(t) ? t[~~(Math.random() * t.length)] : (i = i || 1e-5) && (r = i < 1 ? Math.pow(10, (i + "").length - 2) : 1) && Math.floor(Math.round((t - i / 2 + Math.random() * (e - t + i * 0.99)) / i) * i * r) / r;
  });
}, Br = function() {
  for (var t = arguments.length, e = new Array(t), i = 0; i < t; i++)
    e[i] = arguments[i];
  return function(r) {
    return e.reduce(function(n, s) {
      return s(n);
    }, r);
  };
}, Vr = function(t, e) {
  return function(i) {
    return t(parseFloat(i)) + (e || ut(i));
  };
}, Gr = function(t, e, i) {
  return ir(t, e, 0, 1, i);
}, rr = function(t, e, i) {
  return ot(i, function(r) {
    return t[~~e(r)];
  });
}, Wr = function u(t, e, i) {
  var r = e - t;
  return L(t) ? rr(t, u(0, t.length), e) : ot(i, function(n) {
    return (r + (n - t) % r) % r + t;
  });
}, Qr = function u(t, e, i) {
  var r = e - t, n = r * 2;
  return L(t) ? rr(t, u(0, t.length - 1), e) : ot(i, function(s) {
    return s = (n + (s - t) % n) % n || 0, t + (s > r ? n - s : s);
  });
}, Te = function(t) {
  return t.replace(vr, function(e) {
    var i = e.indexOf("[") + 1, r = e.substring(i || 7, i ? e.indexOf("]") : e.length - 1).split(xr);
    return er(i ? r : +r[0], i ? 0 : +r[1], +r[2] || 1e-5);
  });
}, ir = function(t, e, i, r, n) {
  var s = e - t, a = r - i;
  return ot(n, function(o) {
    return i + ((o - t) / s * a || 0);
  });
}, Yr = function u(t, e, i, r) {
  var n = isNaN(t + e) ? 0 : function(d) {
    return (1 - d) * t + d * e;
  };
  if (!n) {
    var s = I(t), a = {}, o, _, h, f, l;
    if (i === !0 && (r = 1) && (i = null), s)
      t = {
        p: t
      }, e = {
        p: e
      };
    else if (L(t) && !L(e)) {
      for (h = [], f = t.length, l = f - 2, _ = 1; _ < f; _++)
        h.push(u(t[_ - 1], t[_]));
      f--, n = function(g) {
        g *= f;
        var c = Math.min(l, ~~g);
        return h[c](g - c);
      }, i = e;
    } else r || (t = vt(L(t) ? [] : {}, t));
    if (!h) {
      for (o in e)
        ve.call(a, t, o, "get", e[o]);
      n = function(g) {
        return be(g, a) || (s ? t.p : t);
      };
    }
  }
  return ot(i, n);
}, Me = function(t, e, i) {
  var r = t.labels, n = K, s, a, o;
  for (s in r)
    a = r[s] - e, a < 0 == !!i && a && n > (a = Math.abs(a)) && (o = s, n = a);
  return o;
}, B = function(t, e, i) {
  var r = t.vars, n = r[e], s = S, a = t._ctx, o, _, h;
  if (n)
    return o = r[e + "Params"], _ = r.callbackScope || t, i && st.length && Vt(), a && (S = a), h = o ? n.apply(_, o) : n.call(_), S = s, h;
}, Dt = function(t) {
  return at(t), t.scrollTrigger && t.scrollTrigger.kill(!!F), t.progress() < 1 && B(t, "onInterrupt"), t;
}, mt, nr = [], sr = function(t) {
  if (t)
    if (t = !t.name && t.default || t, de() || t.headless) {
      var e = t.name, i = O(t), r = e && !i && t.init ? function() {
        this._props = [];
      } : t, n = {
        init: Et,
        render: be,
        add: ve,
        kill: di,
        modifier: li,
        rawVars: 0
      }, s = {
        targetTest: 0,
        get: 0,
        getSetter: lr,
        aliases: {},
        register: 0
      };
      if (bt(), t !== r) {
        if (Y[e])
          return;
        G(r, G(Gt(t, n), s)), vt(r.prototype, vt(n, Gt(t, s))), Y[r.prop = e] = r, t.targetTest && (Ut.push(r), pe[e] = 1), e = (e === "css" ? "CSS" : e.charAt(0).toUpperCase() + e.substr(1)) + "Plugin";
      }
      Be(e, r), t.register && t.register(ct, r, wt);
    } else
      nr.push(t);
}, P = 255, Rt = {
  aqua: [0, P, P],
  lime: [0, P, 0],
  silver: [192, 192, 192],
  black: [0, 0, 0],
  maroon: [128, 0, 0],
  teal: [0, 128, 128],
  blue: [0, 0, P],
  navy: [0, 0, 128],
  white: [P, P, P],
  olive: [128, 128, 0],
  yellow: [P, P, 0],
  orange: [P, 165, 0],
  gray: [128, 128, 128],
  purple: [128, 0, 128],
  green: [0, 128, 0],
  red: [P, 0, 0],
  pink: [P, 192, 203],
  cyan: [0, P, P],
  transparent: [P, P, P, 0]
}, Ht = function(t, e, i) {
  return t += t < 0 ? 1 : t > 1 ? -1 : 0, (t * 6 < 1 ? e + (i - e) * t * 6 : t < 0.5 ? i : t * 3 < 2 ? e + (i - e) * (2 / 3 - t) * 6 : e) * P + 0.5 | 0;
}, ar = function(t, e, i) {
  var r = t ? et(t) ? [t >> 16, t >> 8 & P, t & P] : 0 : Rt.black, n, s, a, o, _, h, f, l, d, g;
  if (!r) {
    if (t.substr(-1) === "," && (t = t.substr(0, t.length - 1)), Rt[t])
      r = Rt[t];
    else if (t.charAt(0) === "#") {
      if (t.length < 6 && (n = t.charAt(1), s = t.charAt(2), a = t.charAt(3), t = "#" + n + n + s + s + a + a + (t.length === 5 ? t.charAt(4) + t.charAt(4) : "")), t.length === 9)
        return r = parseInt(t.substr(1, 6), 16), [r >> 16, r >> 8 & P, r & P, parseInt(t.substr(7), 16) / 255];
      t = parseInt(t.substr(1), 16), r = [t >> 16, t >> 8 & P, t & P];
    } else if (t.substr(0, 3) === "hsl") {
      if (r = g = t.match(De), !e)
        o = +r[0] % 360 / 360, _ = +r[1] / 100, h = +r[2] / 100, s = h <= 0.5 ? h * (_ + 1) : h + _ - h * _, n = h * 2 - s, r.length > 3 && (r[3] *= 1), r[0] = Ht(o + 1 / 3, n, s), r[1] = Ht(o, n, s), r[2] = Ht(o - 1 / 3, n, s);
      else if (~t.indexOf("="))
        return r = t.match(kr), i && r.length < 4 && (r[3] = 1), r;
    } else
      r = t.match(De) || Rt.transparent;
    r = r.map(Number);
  }
  return e && !g && (n = r[0] / P, s = r[1] / P, a = r[2] / P, f = Math.max(n, s, a), l = Math.min(n, s, a), h = (f + l) / 2, f === l ? o = _ = 0 : (d = f - l, _ = h > 0.5 ? d / (2 - f - l) : d / (f + l), o = f === n ? (s - a) / d + (s < a ? 6 : 0) : f === s ? (a - n) / d + 2 : (n - s) / d + 4, o *= 60), r[0] = ~~(o + 0.5), r[1] = ~~(_ * 100 + 0.5), r[2] = ~~(h * 100 + 0.5)), i && r.length < 4 && (r[3] = 1), r;
}, ur = function(t) {
  var e = [], i = [], r = -1;
  return t.split(yt).forEach(function(n) {
    var s = n.match(Ie) || [];
    e.push.apply(e, s), i.push(r += s.length + 1);
  }), e.c = i, e;
}, Ae = function(t, e, i) {
  var r = "", n = (t + r).match(yt), s = e ? "hsla(" : "rgba(", a = 0, o, _, h, f;
  if (!n)
    return t;
  if (n = n.map(function(l) {
    return (l = ar(l, e, 1)) && s + (e ? l[0] + "," + l[1] + "%," + l[2] + "%," + l[3] : l.join(",")) + ")";
  }), i && (h = ur(t), o = i.c, o.join(r) !== h.c.join(r)))
    for (_ = t.replace(yt, "1").split(Ie), f = _.length - 1; a < f; a++)
      r += _[a] + (~o.indexOf(a) ? n.shift() || s + "0,0,0,0)" : (h.length ? h : n.length ? n : i).shift());
  if (!_)
    for (_ = t.split(yt), f = _.length - 1; a < f; a++)
      r += _[a] + n[a];
  return r + _[f];
}, yt = (function() {
  var u = "(?:\\b(?:(?:rgb|rgba|hsl|hsla)\\(.+?\\))|\\B#(?:[0-9a-f]{3,4}){1,2}\\b", t;
  for (t in Rt)
    u += "|" + t + "\\b";
  return new RegExp(u + ")", "gi");
})(), Kr = /hsl[a]?\(/, Xr = function(t) {
  var e = t.join(" "), i;
  if (yt.lastIndex = 0, yt.test(e))
    return i = Kr.test(e), t[1] = Ae(t[1], i), t[0] = Ae(t[0], i, ur(t[1])), !0;
}, zt, Z = (function() {
  var u = Date.now, t = 500, e = 33, i = u(), r = i, n = 1e3 / 240, s = n, a = [], o, _, h, f, l, d, g = function c(p) {
    var m = u() - r, y = p === !0, v, x, T, D;
    if ((m > t || m < 0) && (i += m - e), r += m, T = r - i, v = T - s, (v > 0 || y) && (D = ++f.frame, l = T - f.time * 1e3, f.time = T = T / 1e3, s += v + (v >= n ? 4 : n - v), x = 1), y || (o = _(c)), x)
      for (d = 0; d < a.length; d++)
        a[d](T, l, D, p);
  };
  return f = {
    time: 0,
    frame: 0,
    tick: function() {
      g(!0);
    },
    deltaRatio: function(p) {
      return l / (1e3 / (p || 60));
    },
    wake: function() {
      Ne && (!ie && de() && ($ = ie = window, ce = $.document || {}, V.gsap = ct, ($.gsapVersions || ($.gsapVersions = [])).push(ct.version), Ue(Bt || $.GreenSockGlobals || !$.gsap && $ || {}), nr.forEach(sr)), h = typeof requestAnimationFrame < "u" && requestAnimationFrame, o && f.sleep(), _ = h || function(p) {
        return setTimeout(p, s - f.time * 1e3 + 1 | 0);
      }, zt = 1, g(2));
    },
    sleep: function() {
      (h ? cancelAnimationFrame : clearTimeout)(o), zt = 0, _ = Et;
    },
    lagSmoothing: function(p, m) {
      t = p || 1 / 0, e = Math.min(m || 33, t);
    },
    fps: function(p) {
      n = 1e3 / (p || 240), s = f.time * 1e3 + n;
    },
    add: function(p, m, y) {
      var v = m ? function(x, T, D, k) {
        p(x, T, D, k), f.remove(v);
      } : p;
      return f.remove(p), a[y ? "unshift" : "push"](v), bt(), v;
    },
    remove: function(p, m) {
      ~(m = a.indexOf(p)) && a.splice(m, 1) && d >= m && d--;
    },
    _listeners: a
  }, f;
})(), bt = function() {
  return !zt && Z.wake();
}, b = {}, Zr = /^[\d.\-M][\d.\-,\s]/, $r = /["']/g, Hr = function(t) {
  for (var e = {}, i = t.substr(1, t.length - 3).split(":"), r = i[0], n = 1, s = i.length, a, o, _; n < s; n++)
    o = i[n], a = n !== s - 1 ? o.lastIndexOf(",") : o.length, _ = o.substr(0, a), e[r] = isNaN(_) ? _.replace($r, "").trim() : +_, r = o.substr(a + 1).trim();
  return e;
}, jr = function(t) {
  var e = t.indexOf("(") + 1, i = t.indexOf(")"), r = t.indexOf("(", e);
  return t.substring(e, ~r && r < i ? t.indexOf(")", i + 1) : i);
}, Jr = function(t) {
  var e = (t + "").split("("), i = b[e[0]];
  return i && e.length > 1 && i.config ? i.config.apply(null, ~t.indexOf("{") ? [Hr(e[1])] : jr(t).split(",").map(Qe)) : b._CE && Zr.test(t) ? b._CE("", t) : i;
}, ti = function(t) {
  return function(e) {
    return 1 - t(1 - e);
  };
}, lt = function(t, e) {
  return t && (O(t) ? t : b[t] || Jr(t)) || e;
}, pt = function(t, e, i, r) {
  i === void 0 && (i = function(o) {
    return 1 - e(1 - o);
  }), r === void 0 && (r = function(o) {
    return o < 0.5 ? e(o * 2) / 2 : 1 - e((1 - o) * 2) / 2;
  });
  var n = {
    easeIn: e,
    easeOut: i,
    easeInOut: r
  }, s;
  return rt(t, function(a) {
    b[a] = V[a] = n, b[s = a.toLowerCase()] = i;
    for (var o in n)
      b[s + (o === "easeIn" ? ".in" : o === "easeOut" ? ".out" : ".inOut")] = b[a + "." + o] = n[o];
  }), n;
}, or = function(t) {
  return function(e) {
    return e < 0.5 ? (1 - t(1 - e * 2)) / 2 : 0.5 + t((e - 0.5) * 2) / 2;
  };
}, jt = function u(t, e, i) {
  var r = e >= 1 ? e : 1, n = (i || (t ? 0.3 : 0.45)) / (e < 1 ? e : 1), s = n / re * (Math.asin(1 / r) || 0), a = function(h) {
    return h === 1 ? 1 : r * Math.pow(2, -10 * h) * Tr((h - s) * n) + 1;
  }, o = t === "out" ? a : t === "in" ? function(_) {
    return 1 - a(1 - _);
  } : or(a);
  return n = re / n, o.config = function(_, h) {
    return u(t, _, h);
  }, o;
}, Jt = function u(t, e) {
  e === void 0 && (e = 1.70158);
  var i = function(s) {
    return s ? --s * s * ((e + 1) * s + e) + 1 : 0;
  }, r = t === "out" ? i : t === "in" ? function(n) {
    return 1 - i(1 - n);
  } : or(i);
  return r.config = function(n) {
    return u(t, n);
  }, r;
};
rt("Linear,Quad,Cubic,Quart,Quint,Strong", function(u, t) {
  var e = t < 5 ? t + 1 : t;
  pt(u + ",Power" + (e - 1), t ? function(i) {
    return Math.pow(i, e);
  } : function(i) {
    return i;
  }, function(i) {
    return 1 - Math.pow(1 - i, e);
  }, function(i) {
    return i < 0.5 ? Math.pow(i * 2, e) / 2 : 1 - Math.pow((1 - i) * 2, e) / 2;
  });
});
b.Linear.easeNone = b.none = b.Linear.easeIn;
pt("Elastic", jt("in"), jt("out"), jt());
(function(u, t) {
  var e = 1 / t, i = 2 * e, r = 2.5 * e, n = function(a) {
    return a < e ? u * a * a : a < i ? u * Math.pow(a - 1.5 / t, 2) + 0.75 : a < r ? u * (a -= 2.25 / t) * a + 0.9375 : u * Math.pow(a - 2.625 / t, 2) + 0.984375;
  };
  pt("Bounce", function(s) {
    return 1 - n(1 - s);
  }, n);
})(7.5625, 2.75);
pt("Expo", function(u) {
  return Math.pow(2, 10 * (u - 1)) * u + u * u * u * u * u * u * (1 - u);
});
pt("Circ", function(u) {
  return -(ze(1 - u * u) - 1);
});
pt("Sine", function(u) {
  return u === 1 ? 1 : -yr(u * gr) + 1;
});
pt("Back", Jt("in"), Jt("out"), Jt());
b.SteppedEase = b.steps = V.SteppedEase = {
  config: function(t, e) {
    t === void 0 && (t = 1);
    var i = 1 / t, r = t + (e ? 0 : 1), n = e ? 1 : 0, s = 1 - C;
    return function(a) {
      return ((r * Lt(0, s, a) | 0) + n) * i;
    };
  }
};
At.ease = b["quad.out"];
rt("onComplete,onUpdate,onStart,onRepeat,onReverseComplete,onInterrupt", function(u) {
  return ge += u + "," + u + "Params,";
});
var ei = function(t, e) {
  this.id = mr++, t._gsap = this, this.target = t, this.harness = e, this.get = e ? e.get : Cr, this.set = e ? e.getSetter : lr;
}, Ft = /* @__PURE__ */ (function() {
  function u(e) {
    this.vars = e, this._delay = +e.delay || 0, (this._repeat = e.repeat === 1 / 0 ? -2 : e.repeat || 0) && (this._rDelay = e.repeatDelay || 0, this._yoyo = !!e.yoyo || !!e.yoyoEase), this._ts = 1, kt(this, +e.duration, 1, 1), this.data = e.data, S && (this._ctx = S, S.data.push(this)), zt || Z.wake();
  }
  var t = u.prototype;
  return t.delay = function(i) {
    return i || i === 0 ? (this.parent && this.parent.smoothChildTiming && this.startTime(this._start + i - this._delay), this._delay = i, this) : this._delay;
  }, t.duration = function(i) {
    return arguments.length ? this.totalDuration(this._repeat > 0 ? i + (i + this._rDelay) * this._repeat : i) : this.totalDuration() && this._dur;
  }, t.totalDuration = function(i) {
    return arguments.length ? (this._dirty = 0, kt(this, this._repeat < 0 ? i : (i - this._repeat * this._rDelay) / (this._repeat + 1))) : this._tDur;
  }, t.totalTime = function(i, r) {
    if (bt(), !arguments.length)
      return this._tTime;
    var n = this._dp;
    if (n && n.smoothChildTiming && this._ts) {
      for (Xt(this, i), !n._dp || n.parent || Xe(n, this); n && n.parent; )
        n.parent._time !== n._start + (n._ts >= 0 ? n._tTime / n._ts : (n.totalDuration() - n._tTime) / -n._ts) && n.totalTime(n._tTime, !0), n = n.parent;
      !this.parent && this._dp.autoRemoveChildren && (this._ts > 0 && i < this._tDur || this._ts < 0 && i > 0 || !this._tDur && !i) && H(this._dp, this, this._start - this._delay);
    }
    return (this._tTime !== i || !this._dur && !r || this._initted && Math.abs(this._zTime) === C || !this._initted && this._dur && i || !i && !this._initted && (this.add || this._ptLookup)) && (this._ts || (this._pTime = i), We(this, i, r)), this;
  }, t.time = function(i, r) {
    return arguments.length ? this.totalTime(Math.min(this.totalDuration(), i + Ce(this)) % (this._dur + this._rDelay) || (i ? this._dur : 0), r) : this._time;
  }, t.totalProgress = function(i, r) {
    return arguments.length ? this.totalTime(this.totalDuration() * i, r) : this.totalDuration() ? Math.min(1, this._tTime / this._tDur) : this.rawTime() >= 0 && this._initted ? 1 : 0;
  }, t.progress = function(i, r) {
    return arguments.length ? this.totalTime(this.duration() * (this._yoyo && !(this.iteration() & 1) ? 1 - i : i) + Ce(this), r) : this.duration() ? Math.min(1, this._time / this._dur) : this.rawTime() > 0 ? 1 : 0;
  }, t.iteration = function(i, r) {
    var n = this.duration() + this._rDelay;
    return arguments.length ? this.totalTime(this._time + (i - 1) * n, r) : this._repeat ? xt(this._tTime, n) + 1 : 1;
  }, t.timeScale = function(i, r) {
    if (!arguments.length)
      return this._rts === -C ? 0 : this._rts;
    if (this._rts === i)
      return this;
    var n = this.parent && this._ts ? Wt(this.parent._time, this) : this._tTime;
    return this._rts = +i || 0, this._ts = this._ps || i === -C ? 0 : this._rts, this.totalTime(Lt(-Math.abs(this._delay), this.totalDuration(), n), r !== !1), Kt(this), Er(this);
  }, t.paused = function(i) {
    return arguments.length ? (this._ps !== i && (this._ps = i, i ? (this._pTime = this._tTime || Math.max(-this._delay, this.rawTime()), this._ts = this._act = 0) : (bt(), this._ts = this._rts, this.totalTime(this.parent && !this.parent.smoothChildTiming ? this.rawTime() : this._tTime || this._pTime, this.progress() === 1 && Math.abs(this._zTime) !== C && (this._tTime -= C)))), this) : this._ps;
  }, t.startTime = function(i) {
    if (arguments.length) {
      this._start = M(i);
      var r = this.parent || this._dp;
      return r && (r._sort || !this.parent) && H(r, this, this._start - this._delay), this;
    }
    return this._start;
  }, t.endTime = function(i) {
    return this._start + (q(i) ? this.totalDuration() : this.duration()) / Math.abs(this._ts || 1);
  }, t.rawTime = function(i) {
    var r = this.parent || this._dp;
    return r ? i && (!this._ts || this._repeat && this._time && this.totalProgress() < 1) ? this._tTime % (this._dur + this._rDelay) : this._ts ? Wt(r.rawTime(i), this) : this._tTime : this._tTime;
  }, t.revert = function(i) {
    i === void 0 && (i = Pr);
    var r = F;
    return F = i, ye(this) && (this.timeline && this.timeline.revert(i), this.totalTime(-0.01, i.suppressEvents)), this.data !== "nested" && i.kill !== !1 && this.kill(), F = r, this;
  }, t.globalTime = function(i) {
    for (var r = this, n = arguments.length ? i : r.rawTime(); r; )
      n = r._start + n / (Math.abs(r._ts) || 1), r = r._dp;
    return !this.parent && this._sat ? this._sat.globalTime(i) : n;
  }, t.repeat = function(i) {
    return arguments.length ? (this._repeat = i === 1 / 0 ? -2 : i, Se(this)) : this._repeat === -2 ? 1 / 0 : this._repeat;
  }, t.repeatDelay = function(i) {
    if (arguments.length) {
      var r = this._time;
      return this._rDelay = i, Se(this), r ? this.time(r) : this;
    }
    return this._rDelay;
  }, t.yoyo = function(i) {
    return arguments.length ? (this._yoyo = i, this) : this._yoyo;
  }, t.seek = function(i, r) {
    return this.totalTime(Q(this, i), q(r));
  }, t.restart = function(i, r) {
    return this.play().totalTime(i ? -this._delay : 0, q(r)), this._dur || (this._zTime = -C), this;
  }, t.play = function(i, r) {
    return i != null && this.seek(i, r), this.reversed(!1).paused(!1);
  }, t.reverse = function(i, r) {
    return i != null && this.seek(i || this.totalDuration(), r), this.reversed(!0).paused(!1);
  }, t.pause = function(i, r) {
    return i != null && this.seek(i, r), this.paused(!0);
  }, t.resume = function() {
    return this.paused(!1);
  }, t.reversed = function(i) {
    return arguments.length ? (!!i !== this.reversed() && this.timeScale(-this._rts || (i ? -C : 0)), this) : this._rts < 0;
  }, t.invalidate = function() {
    return this._initted = this._act = 0, this._zTime = -C, this;
  }, t.isActive = function() {
    var i = this.parent || this._dp, r = this._start, n;
    return !!(!i || this._ts && this._initted && i.isActive() && (n = i.rawTime(!0)) >= r && n < this.endTime(!0) - C);
  }, t.eventCallback = function(i, r, n) {
    var s = this.vars;
    return arguments.length > 1 ? (r ? (s[i] = r, n && (s[i + "Params"] = n), i === "onUpdate" && (this._onUpdate = r)) : delete s[i], this) : s[i];
  }, t.then = function(i) {
    var r = this, n = r._prom;
    return new Promise(function(s) {
      var a = O(i) ? i : Ye, o = function() {
        var h = r.then;
        r.then = null, n && n(), O(a) && (a = a(r)) && (a.then || a === r) && (r.then = h), s(a), r.then = h;
      };
      r._initted && r.totalProgress() === 1 && r._ts >= 0 || !r._tTime && r._ts < 0 ? o() : r._prom = o;
    });
  }, t.kill = function() {
    Dt(this);
  }, u;
})();
G(Ft.prototype, {
  _time: 0,
  _start: 0,
  _end: 0,
  _tTime: 0,
  _tDur: 0,
  _dirty: 0,
  _repeat: 0,
  _yoyo: !1,
  parent: null,
  _initted: !1,
  _rDelay: 0,
  _ts: 1,
  _dp: 0,
  ratio: 0,
  _zTime: -C,
  _prom: 0,
  _ps: !1,
  _rts: 1
});
var U = /* @__PURE__ */ (function(u) {
  Ee(t, u);
  function t(i, r) {
    var n;
    return i === void 0 && (i = {}), n = u.call(this, i) || this, n.labels = {}, n.smoothChildTiming = !!i.smoothChildTiming, n.autoRemoveChildren = !!i.autoRemoveChildren, n._sort = q(i.sortChildren), A && H(i.parent || A, tt(n), r), i.reversed && n.reverse(), i.paused && n.paused(!0), i.scrollTrigger && Ze(tt(n), i.scrollTrigger), n;
  }
  var e = t.prototype;
  return e.to = function(r, n, s) {
    return St(0, arguments, this), this;
  }, e.from = function(r, n, s) {
    return St(1, arguments, this), this;
  }, e.fromTo = function(r, n, s, a) {
    return St(2, arguments, this), this;
  }, e.set = function(r, n, s) {
    return n.duration = 0, n.parent = this, Ct(n).repeatDelay || (n.repeat = 0), n.immediateRender = !!n.immediateRender, new z(r, n, Q(this, s), 1), this;
  }, e.call = function(r, n, s) {
    return H(this, z.delayedCall(0, r, n), s);
  }, e.staggerTo = function(r, n, s, a, o, _, h) {
    return s.duration = n, s.stagger = s.stagger || a, s.onComplete = _, s.onCompleteParams = h, s.parent = this, new z(r, s, Q(this, o)), this;
  }, e.staggerFrom = function(r, n, s, a, o, _, h) {
    return s.runBackwards = 1, Ct(s).immediateRender = q(s.immediateRender), this.staggerTo(r, n, s, a, o, _, h);
  }, e.staggerFromTo = function(r, n, s, a, o, _, h, f) {
    return a.startAt = s, Ct(a).immediateRender = q(a.immediateRender), this.staggerTo(r, n, a, o, _, h, f);
  }, e.render = function(r, n, s) {
    var a = this._time, o = this._dirty ? this.totalDuration() : this._tDur, _ = this._dur, h = r <= 0 ? 0 : M(r), f = this._zTime < 0 != r < 0 && (this._initted || !_), l, d, g, c, p, m, y, v, x, T, D, k;
    if (this !== A && h > o && r >= 0 && (h = o), h !== this._tTime || s || f) {
      if (a !== this._time && _ && (h += this._time - a, r += this._time - a), l = h, x = this._start, v = this._ts, m = !v, f && (_ || (a = this._zTime), (r || !n) && (this._zTime = r)), this._repeat) {
        if (D = this._yoyo, p = _ + this._rDelay, this._repeat < -1 && r < 0)
          return this.totalTime(p * 100 + r, n, s);
        if (l = M(h % p), h === o ? (c = this._repeat, l = _) : (T = M(h / p), c = ~~T, c && c === T && (l = _, c--), l > _ && (l = _)), T = xt(this._tTime, p), !a && this._tTime && T !== c && this._tTime - T * p - this._dur <= 0 && (T = c), D && c & 1 && (l = _ - l, k = 1), c !== T && !this._lock) {
          var R = D && T & 1, w = R === (D && c & 1);
          if (c < T && (R = !R), a = R ? 0 : h % _ ? _ : h, this._lock = 1, this.render(a || (k ? 0 : M(c * p)), n, !_)._lock = 0, this._tTime = h, !n && this.parent && B(this, "onRepeat"), this.vars.repeatRefresh && !k && (this.invalidate()._lock = 1, T = c), a && a !== this._time || m !== !this._ts || this.vars.onRepeat && !this.parent && !this._act)
            return this;
          if (_ = this._dur, o = this._tDur, w && (this._lock = 2, a = R ? _ : -1e-4, this.render(a, !0), this.vars.repeatRefresh && !k && this.invalidate()), this._lock = 0, !this._ts && !m)
            return this;
        }
      }
      if (this._hasPause && !this._forcing && this._lock < 2 && (y = Ir(this, M(a), M(l)), y && (h -= l - (l = y._start))), this._tTime = h, this._time = l, this._act = !!v, this._initted || (this._onUpdate = this.vars.onUpdate, this._initted = 1, this._zTime = r, a = 0), !a && h && _ && !n && !T && (B(this, "onStart"), this._tTime !== h))
        return this;
      if (l >= a && r >= 0)
        for (d = this._first; d; ) {
          if (g = d._next, (d._act || l >= d._start) && d._ts && y !== d) {
            if (d.parent !== this)
              return this.render(r, n, s);
            if (d.render(d._ts > 0 ? (l - d._start) * d._ts : (d._dirty ? d.totalDuration() : d._tDur) + (l - d._start) * d._ts, n, s), l !== this._time || !this._ts && !m) {
              y = 0, g && (h += this._zTime = -C);
              break;
            }
          }
          d = g;
        }
      else {
        d = this._last;
        for (var E = r < 0 ? r : l; d; ) {
          if (g = d._prev, (d._act || E <= d._end) && d._ts && y !== d) {
            if (d.parent !== this)
              return this.render(r, n, s);
            if (d.render(d._ts > 0 ? (E - d._start) * d._ts : (d._dirty ? d.totalDuration() : d._tDur) + (E - d._start) * d._ts, n, s || F && ye(d)), l !== this._time || !this._ts && !m) {
              y = 0, g && (h += this._zTime = E ? -C : C);
              break;
            }
          }
          d = g;
        }
      }
      if (y && !n && (this.pause(), y.render(l >= a ? 0 : -C)._zTime = l >= a ? 1 : -1, this._ts))
        return this._start = x, Kt(this), this.render(r, n, s);
      this._onUpdate && !n && B(this, "onUpdate", !0), (h === o && this._tTime >= this.totalDuration() || !h && a) && (x === this._start || Math.abs(v) !== Math.abs(this._ts)) && (this._lock || ((r || !_) && (h === o && this._ts > 0 || !h && this._ts < 0) && at(this, 1), !n && !(r < 0 && !a) && (h || a || !o) && (B(this, h === o && r >= 0 ? "onComplete" : "onReverseComplete", !0), this._prom && !(h < o && this.timeScale() > 0) && this._prom())));
    }
    return this;
  }, e.add = function(r, n) {
    var s = this;
    if (et(n) || (n = Q(this, n, r)), !(r instanceof Ft)) {
      if (L(r))
        return r.forEach(function(a) {
          return s.add(a, n);
        }), this;
      if (I(r))
        return this.addLabel(r, n);
      if (O(r))
        r = z.delayedCall(0, r);
      else
        return this;
    }
    return this !== r ? H(this, r, n) : this;
  }, e.getChildren = function(r, n, s, a) {
    r === void 0 && (r = !0), n === void 0 && (n = !0), s === void 0 && (s = !0), a === void 0 && (a = -K);
    for (var o = [], _ = this._first; _; )
      _._start >= a && (_ instanceof z ? n && o.push(_) : (s && o.push(_), r && o.push.apply(o, _.getChildren(!0, n, s)))), _ = _._next;
    return o;
  }, e.getById = function(r) {
    for (var n = this.getChildren(1, 1, 1), s = n.length; s--; )
      if (n[s].vars.id === r)
        return n[s];
  }, e.remove = function(r) {
    return I(r) ? this.removeLabel(r) : O(r) ? this.killTweensOf(r) : (r.parent === this && Yt(this, r), r === this._recent && (this._recent = this._last), ft(this));
  }, e.totalTime = function(r, n) {
    return arguments.length ? (this._forcing = 1, !this._dp && this._ts && (this._start = M(Z.time - (this._ts > 0 ? r / this._ts : (this.totalDuration() - r) / -this._ts))), u.prototype.totalTime.call(this, r, n), this._forcing = 0, this) : this._tTime;
  }, e.addLabel = function(r, n) {
    return this.labels[r] = Q(this, n), this;
  }, e.removeLabel = function(r) {
    return delete this.labels[r], this;
  }, e.addPause = function(r, n, s) {
    var a = z.delayedCall(0, n || Et, s);
    return a.data = "isPause", this._hasPause = 1, H(this, a, Q(this, r));
  }, e.removePause = function(r) {
    var n = this._first;
    for (r = Q(this, r); n; )
      n._start === r && n.data === "isPause" && at(n), n = n._next;
  }, e.killTweensOf = function(r, n, s) {
    for (var a = this.getTweensOf(r, s), o = a.length; o--; )
      nt !== a[o] && a[o].kill(r, n);
    return this;
  }, e.getTweensOf = function(r, n) {
    for (var s = [], a = X(r), o = this._first, _ = et(n), h; o; )
      o instanceof z ? Mr(o._targets, a) && (_ ? (!nt || o._initted && o._ts) && o.globalTime(0) <= n && o.globalTime(o.totalDuration()) > n : !n || o.isActive()) && s.push(o) : (h = o.getTweensOf(a, n)).length && s.push.apply(s, h), o = o._next;
    return s;
  }, e.tweenTo = function(r, n) {
    n = n || {};
    var s = this, a = Q(s, r), o = n, _ = o.startAt, h = o.onStart, f = o.onStartParams, l = o.immediateRender, d, g = z.to(s, G({
      ease: n.ease || "none",
      lazy: !1,
      immediateRender: !1,
      time: a,
      overwrite: "auto",
      duration: n.duration || Math.abs((a - (_ && "time" in _ ? _.time : s._time)) / s.timeScale()) || C,
      onStart: function() {
        if (s.pause(), !d) {
          var p = n.duration || Math.abs((a - (_ && "time" in _ ? _.time : s._time)) / s.timeScale());
          g._dur !== p && kt(g, p, 0, 1).render(g._time, !0, !0), d = 1;
        }
        h && h.apply(g, f || []);
      }
    }, n));
    return l ? g.render(0) : g;
  }, e.tweenFromTo = function(r, n, s) {
    return this.tweenTo(n, G({
      startAt: {
        time: Q(this, r)
      }
    }, s));
  }, e.recent = function() {
    return this._recent;
  }, e.nextLabel = function(r) {
    return r === void 0 && (r = this._time), Me(this, Q(this, r));
  }, e.previousLabel = function(r) {
    return r === void 0 && (r = this._time), Me(this, Q(this, r), 1);
  }, e.currentLabel = function(r) {
    return arguments.length ? this.seek(r, !0) : this.previousLabel(this._time + C);
  }, e.shiftChildren = function(r, n, s) {
    s === void 0 && (s = 0);
    var a = this._first, o = this.labels, _;
    for (r = M(r); a; )
      a._start >= s && (a._start += r, a._end += r), a = a._next;
    if (n)
      for (_ in o)
        o[_] >= s && (o[_] += r);
    return ft(this);
  }, e.invalidate = function(r) {
    var n = this._first;
    for (this._lock = 0; n; )
      n.invalidate(r), n = n._next;
    return u.prototype.invalidate.call(this, r);
  }, e.clear = function(r) {
    r === void 0 && (r = !0);
    for (var n = this._first, s; n; )
      s = n._next, this.remove(n), n = s;
    return this._dp && (this._time = this._tTime = this._pTime = 0), r && (this.labels = {}), ft(this);
  }, e.totalDuration = function(r) {
    var n = 0, s = this, a = s._last, o = K, _, h, f;
    if (arguments.length)
      return s.timeScale((s._repeat < 0 ? s.duration() : s.totalDuration()) / (s.reversed() ? -r : r));
    if (s._dirty) {
      for (f = s.parent; a; )
        _ = a._prev, a._dirty && a.totalDuration(), h = a._start, h > o && s._sort && a._ts && !s._lock ? (s._lock = 1, H(s, a, h - a._delay, 1)._lock = 0) : o = h, h < 0 && a._ts && (n -= h, (!f && !s._dp || f && f.smoothChildTiming) && (s._start += M(h / s._ts), s._time -= h, s._tTime -= h), s.shiftChildren(-h, !1, -1 / 0), o = 0), a._end > n && a._ts && (n = a._end), a = _;
      kt(s, s === A && s._time > n ? s._time : n, 1, 1), s._dirty = 0;
    }
    return s._tDur;
  }, t.updateRoot = function(r) {
    if (A._ts && (We(A, Wt(r, A)), Ve = Z.frame), Z.frame >= Re) {
      Re += Tt.autoSleep || 120;
      var n = A._first;
      if ((!n || !n._ts) && Tt.autoSleep && Z._listeners.length < 2) {
        for (; n && !n._ts; )
          n = n._next;
        n || Z.sleep();
      }
    }
  }, t;
})(Ft);
G(U.prototype, {
  _lock: 0,
  _hasPause: 0,
  _forcing: 0
});
var ri = function(t, e, i, r, n, s, a) {
  var o = new wt(this._pt, t, e, 0, 1, fi, null, n), _ = 0, h = 0, f, l, d, g, c, p, m, y;
  for (o.b = i, o.e = r, i += "", r += "", (m = ~r.indexOf("random(")) && (r = Te(r)), s && (y = [i, r], s(y, t, e), i = y[0], r = y[1]), l = i.match(Zt) || []; f = Zt.exec(r); )
    g = f[0], c = r.substring(_, f.index), d ? d = (d + 1) % 5 : c.substr(-5) === "rgba(" && (d = 1), g !== l[h++] && (p = parseFloat(l[h - 1]) || 0, o._pt = {
      _next: o._pt,
      p: c || h === 1 ? c : ",",
      //note: SVG spec allows omission of comma/space when a negative sign is wedged between two numbers, like 2.5-5.3 instead of 2.5,-5.3 but when tweening, the negative value may switch to positive, so we insert the comma just in case.
      s: p,
      c: g.charAt(1) === "=" ? Ge(p, g) - p : parseFloat(g) - p,
      m: d && d < 4 ? Math.round : 0
    }, _ = Zt.lastIndex);
  return o.c = _ < r.length ? r.substring(_, r.length) : "", o.fp = a, (br.test(r) || m) && (o.e = 0), this._pt = o, o;
}, ve = function(t, e, i, r, n, s, a, o, _, h) {
  O(r) && (r = r(n || 0, t, s));
  var f = t[e], l = i !== "get" ? i : O(f) ? _ ? t[e.indexOf("set") || !O(t["get" + e.substr(3)]) ? e : "get" + e.substr(3)](_) : t[e]() : f, d = O(f) ? _ ? oi : fr : ke, g;
  if (I(r) && (~r.indexOf("random(") && (r = Te(r)), r.charAt(1) === "=" && (g = Ge(l, r) + (ut(l) || 0), (g || g === 0) && (r = g))), !h || l !== r || he)
    return !isNaN(l * r) && r !== "" ? (g = new wt(this._pt, t, e, +l || 0, r - (l || 0), typeof f == "boolean" ? hi : dr, 0, d), _ && (g.fp = _), a && g.modifier(a, this, t), this._pt = g) : (!f && !(e in t) && qe(e, r), ri.call(this, t, e, l, r, d, o || Tt.stringFilter, _));
}, ii = function(t, e, i, r, n) {
  if (O(t) && (t = Mt(t, n, e, i, r)), !j(t) || t.style && t.nodeType || L(t) || Le(t))
    return I(t) ? Mt(t, n, e, i, r) : t;
  var s = {}, a;
  for (a in t)
    s[a] = Mt(t[a], n, e, i, r);
  return s;
}, ni = function(t, e, i, r, n, s) {
  var a, o, _, h;
  if (Y[t] && (a = new Y[t]()).init(n, a.rawVars ? e[t] : ii(e[t], r, n, s, i), i, r, s) !== !1 && (i._pt = o = new wt(i._pt, n, t, 0, 1, a.render, a, 0, a.priority), i !== mt))
    for (_ = i._ptLookup[i._targets.indexOf(n)], h = a._props.length; h--; )
      _[a._props[h]] = o;
  return a;
}, nt, he, xe = function u(t, e, i) {
  var r = t.vars, n = r.ease, s = r.startAt, a = r.immediateRender, o = r.lazy, _ = r.onUpdate, h = r.runBackwards, f = r.yoyoEase, l = r.keyframes, d = r.autoRevert, g = t._dur, c = t._startAt, p = t._targets, m = t.parent, y = m && m.data === "nested" ? m.vars.targets : p, v = t._overwrite === "auto" && !le, x = t.timeline, T = r.easeReverse || f, D, k, R, w, E, gt, _t, W, N, J, it, ht, we;
  if (x && (!l || !n) && (n = "none"), t._ease = lt(n, At.ease), t._rEase = T && (lt(T) || t._ease), t._from = !x && !!r.runBackwards, t._from && (t.ratio = 1), !x || l && !r.stagger) {
    if (W = p[0] ? Pt(p[0]).harness : 0, ht = W && r[W.prop], D = Gt(r, pe), c && (c._zTime < 0 && c.progress(1), e < 0 && h && a && !d ? c.render(-1, !0) : c.revert(h && g ? Nt : Rr), c._lazy = 0), s) {
      if (at(t._startAt = z.set(p, G({
        data: "isStart",
        overwrite: !1,
        parent: m,
        immediateRender: !0,
        lazy: !c && q(o),
        startAt: null,
        delay: 0,
        onUpdate: _ && function() {
          return B(t, "onUpdate");
        },
        stagger: 0
      }, s))), t._startAt._dp = 0, t._startAt._sat = t, e < 0 && (F || !a && !d) && t._startAt.revert(Nt), a && g && e <= 0 && i <= 0) {
        e && (t._zTime = e);
        return;
      }
    } else if (h && g && !c) {
      if (e && (a = !1), R = G({
        overwrite: !1,
        data: "isFromStart",
        //we tag the tween with as "isFromStart" so that if [inside a plugin] we need to only do something at the very END of a tween, we have a way of identifying this tween as merely the one that's setting the beginning values for a "from()" tween. For example, clearProps in CSSPlugin should only get applied at the very END of a tween and without this tag, from(...{height:100, clearProps:"height", delay:1}) would wipe the height at the beginning of the tween and after 1 second, it'd kick back in.
        lazy: a && !c && q(o),
        immediateRender: a,
        //zero-duration tweens render immediately by default, but if we're not specifically instructed to render this tween immediately, we should skip this and merely _init() to record the starting values (rendering them immediately would push them to completion which is wasteful in that case - we'd have to render(-1) immediately after)
        stagger: 0,
        parent: m
        //ensures that nested tweens that had a stagger are handled properly, like gsap.from(".class", {y: gsap.utils.wrap([-100,100]), stagger: 0.5})
      }, D), ht && (R[W.prop] = ht), at(t._startAt = z.set(p, R)), t._startAt._dp = 0, t._startAt._sat = t, e < 0 && (F ? t._startAt.revert(Nt) : t._startAt.render(-1, !0)), t._zTime = e, !a)
        u(t._startAt, C, C);
      else if (!e)
        return;
    }
    for (t._pt = t._ptCache = 0, o = g && q(o) || o && !g, k = 0; k < p.length; k++) {
      if (E = p[k], _t = E._gsap || me(p)[k]._gsap, t._ptLookup[k] = J = {}, ne[_t.id] && st.length && Vt(), it = y === p ? k : y.indexOf(E), W && (N = new W()).init(E, ht || D, t, it, y) !== !1 && (t._pt = w = new wt(t._pt, E, N.name, 0, 1, N.render, N, 0, N.priority), N._props.forEach(function(pr) {
        J[pr] = w;
      }), N.priority && (gt = 1)), !W || ht)
        for (R in D)
          Y[R] && (N = ni(R, D, t, it, E, y)) ? N.priority && (gt = 1) : J[R] = w = ve.call(t, E, R, "get", D[R], it, y, 0, r.stringFilter);
      t._op && t._op[k] && t.kill(E, t._op[k]), v && t._pt && (nt = t, A.killTweensOf(E, J, t.globalTime(e)), we = !t.parent, nt = 0), t._pt && o && (ne[_t.id] = 1);
    }
    gt && pi(t), t._onInit && t._onInit(t);
  }
  t._onUpdate = _, t._initted = (!t._op || t._pt) && !we, l && e <= 0 && x.render(K, !0, !0);
}, si = function(t, e, i, r, n, s, a, o) {
  var _ = (t._pt && t._ptCache || (t._ptCache = {}))[e], h, f, l, d;
  if (!_)
    for (_ = t._ptCache[e] = [], l = t._ptLookup, d = t._targets.length; d--; ) {
      if (h = l[d][e], h && h.d && h.d._pt)
        for (h = h.d._pt; h && h.p !== e && h.fp !== e; )
          h = h._next;
      if (!h)
        return he = 1, t.vars[e] = "+=0", xe(t, a), he = 0, o ? Ot(e + " not eligible for reset. Try splitting into individual properties") : 1;
      _.push(h);
    }
  for (d = _.length; d--; )
    f = _[d], h = f._pt || f, h.s = (r || r === 0) && !n ? r : h.s + (r || 0) + s * h.c, h.c = i - h.s, f.e && (f.e = Sr(i) + ut(f.e)), f.b && (f.b = h.s + ut(f.b));
}, ai = function(t, e) {
  var i = t[0] ? Pt(t[0]).harness : 0, r = i && i.aliases, n, s, a, o;
  if (!r)
    return e;
  n = vt({}, e);
  for (s in r)
    if (s in n)
      for (o = r[s].split(","), a = o.length; a--; )
        n[o[a]] = n[s];
  return n;
}, ui = function(t, e, i, r) {
  var n = e.ease || r || "power1.inOut", s, a;
  if (L(e))
    a = i[t] || (i[t] = []), e.forEach(function(o, _) {
      return a.push({
        t: _ / (e.length - 1) * 100,
        v: o,
        e: n
      });
    });
  else
    for (s in e)
      a = i[s] || (i[s] = []), s === "ease" || a.push({
        t: parseFloat(t),
        v: e[s],
        e: n
      });
}, Mt = function(t, e, i, r, n) {
  return O(t) ? t.call(e, i, r, n) : I(t) && ~t.indexOf("random(") ? Te(t) : t;
}, _r = ge + "repeat,repeatDelay,yoyo,repeatRefresh,yoyoEase,easeReverse,autoRevert", hr = {};
rt(_r + ",id,stagger,delay,duration,paused,scrollTrigger", function(u) {
  return hr[u] = 1;
});
var z = /* @__PURE__ */ (function(u) {
  Ee(t, u);
  function t(i, r, n, s) {
    var a;
    typeof r == "number" && (n.duration = r, r = n, n = null), a = u.call(this, s ? r : Ct(r)) || this;
    var o = a.vars, _ = o.duration, h = o.delay, f = o.immediateRender, l = o.stagger, d = o.overwrite, g = o.keyframes, c = o.defaults, p = o.scrollTrigger, m = r.parent || A, y = (L(i) || Le(i) ? et(i[0]) : "length" in r) ? [i] : X(i), v, x, T, D, k, R, w, E;
    if (a._targets = y.length ? me(y) : Ot("GSAP target " + i + " not found. https://gsap.com", !Tt.nullTargetWarn) || [], a._ptLookup = [], a._overwrite = d, g || l || It(_) || It(h)) {
      r = a.vars;
      var gt = r.easeReverse || r.yoyoEase;
      if (v = a.timeline = new U({
        data: "nested",
        defaults: c || {},
        targets: m && m.data === "nested" ? m.vars.targets : y
      }), v.kill(), v.parent = v._dp = tt(a), v._start = 0, l || It(_) || It(h)) {
        if (D = y.length, w = l && Je(l), j(l))
          for (k in l)
            ~_r.indexOf(k) && (E || (E = {}), E[k] = l[k]);
        for (x = 0; x < D; x++)
          T = Gt(r, hr), T.stagger = 0, gt && (T.easeReverse = gt), E && vt(T, E), R = y[x], T.duration = +Mt(_, tt(a), x, R, y), T.delay = (+Mt(h, tt(a), x, R, y) || 0) - a._delay, !l && D === 1 && T.delay && (a._delay = h = T.delay, a._start += h, T.delay = 0), v.to(R, T, w ? w(x, R, y) : 0), v._ease = b.none;
        v.duration() ? _ = h = 0 : a.timeline = 0;
      } else if (g) {
        Ct(G(v.vars.defaults, {
          ease: "none"
        })), v._ease = lt(g.ease || r.ease || "none");
        var _t = 0, W, N, J;
        if (L(g))
          g.forEach(function(it) {
            return v.to(y, it, ">");
          }), v.duration();
        else {
          T = {};
          for (k in g)
            k === "ease" || k === "easeEach" || ui(k, g[k], T, g.easeEach);
          for (k in T)
            for (W = T[k].sort(function(it, ht) {
              return it.t - ht.t;
            }), _t = 0, x = 0; x < W.length; x++)
              N = W[x], J = {
                ease: N.e,
                duration: (N.t - (x ? W[x - 1].t : 0)) / 100 * _
              }, J[k] = N.v, v.to(y, J, _t), _t += J.duration;
          v.duration() < _ && v.to({}, {
            duration: _ - v.duration()
          });
        }
      }
      _ || a.duration(_ = v.duration());
    } else
      a.timeline = 0;
    return d === !0 && !le && (nt = tt(a), A.killTweensOf(y), nt = 0), H(m, tt(a), n), r.reversed && a.reverse(), r.paused && a.paused(!0), (f || !_ && !g && a._start === M(m._time) && q(f) && zr(tt(a)) && m.data !== "nested") && (a._tTime = -C, a.render(Math.max(0, -h) || 0)), p && Ze(tt(a), p), a;
  }
  var e = t.prototype;
  return e.render = function(r, n, s) {
    var a = this._time, o = this._tDur, _ = this._dur, h = r < 0, f = r > o - C && !h ? o : r < C ? 0 : r, l, d, g, c, p, m, y, v;
    if (!_)
      Lr(this, r, n, s);
    else if (f !== this._tTime || !r || s || !this._initted && this._tTime || this._startAt && this._zTime < 0 !== h || this._lazy) {
      if (l = f, v = this.timeline, this._repeat) {
        if (c = _ + this._rDelay, this._repeat < -1 && h)
          return this.totalTime(c * 100 + r, n, s);
        if (l = M(f % c), f === o ? (g = this._repeat, l = _) : (p = M(f / c), g = ~~p, g && g === p ? (l = _, g--) : l > _ && (l = _)), m = this._yoyo && g & 1, m && (l = _ - l), p = xt(this._tTime, c), l === a && !s && this._initted && g === p)
          return this._tTime = f, this;
        g !== p && this.vars.repeatRefresh && !m && !this._lock && l !== c && this._initted && (this._lock = s = 1, this.render(M(c * g), !0).invalidate()._lock = 0);
      }
      if (!this._initted) {
        if ($e(this, h ? r : l, s, n, f))
          return this._tTime = 0, this;
        if (a !== this._time && !(s && this.vars.repeatRefresh && g !== p))
          return this;
        if (_ !== this._dur)
          return this.render(r, n, s);
      }
      if (this._rEase) {
        var x = l < a;
        if (x !== this._inv) {
          var T = x ? a : _ - a;
          this._inv = x, this._from && (this.ratio = 1 - this.ratio), this._invRatio = this.ratio, this._invTime = a, this._invRecip = T ? (x ? -1 : 1) / T : 0, this._invScale = x ? -this.ratio : 1 - this.ratio, this._invEase = x ? this._rEase : this._ease;
        }
        this.ratio = y = this._invRatio + this._invScale * this._invEase((l - this._invTime) * this._invRecip);
      } else
        this.ratio = y = this._ease(l / _);
      if (this._from && (this.ratio = y = 1 - y), this._tTime = f, this._time = l, !this._act && this._ts && (this._act = 1, this._lazy = 0), !a && f && !n && !p && (B(this, "onStart"), this._tTime !== f))
        return this;
      for (d = this._pt; d; )
        d.r(y, d.d), d = d._next;
      v && v.render(r < 0 ? r : v._dur * v._ease(l / this._dur), n, s) || this._startAt && (this._zTime = r), this._onUpdate && !n && (h && se(this, r, n, s), B(this, "onUpdate")), this._repeat && g !== p && this.vars.onRepeat && !n && this.parent && B(this, "onRepeat"), (f === this._tDur || !f) && this._tTime === f && (h && !this._onUpdate && se(this, r, !0, !0), (r || !_) && (f === this._tDur && this._ts > 0 || !f && this._ts < 0) && at(this, 1), !n && !(h && !a) && (f || a || m) && (B(this, f === o ? "onComplete" : "onReverseComplete", !0), this._prom && !(f < o && this.timeScale() > 0) && this._prom()));
    }
    return this;
  }, e.targets = function() {
    return this._targets;
  }, e.invalidate = function(r) {
    return (!r || !this.vars.runBackwards) && (this._startAt = 0), this._pt = this._op = this._onUpdate = this._lazy = this.ratio = 0, this._ptLookup = [], this.timeline && this.timeline.invalidate(r), u.prototype.invalidate.call(this, r);
  }, e.resetTo = function(r, n, s, a, o) {
    zt || Z.wake(), this._ts || this.play();
    var _ = Math.min(this._dur, (this._dp._time - this._start) * this._ts), h;
    return this._initted || xe(this, _), h = this._ease(_ / this._dur), si(this, r, n, s, a, h, _, o) ? this.resetTo(r, n, s, a, 1) : (Xt(this, 0), this.parent || Ke(this._dp, this, "_first", "_last", this._dp._sort ? "_start" : 0), this.render(0));
  }, e.kill = function(r, n) {
    if (n === void 0 && (n = "all"), !r && (!n || n === "all"))
      return this._lazy = this._pt = 0, this.parent ? Dt(this) : this.scrollTrigger && this.scrollTrigger.kill(!!F), this;
    if (this.timeline) {
      var s = this.timeline.totalDuration();
      return this.timeline.killTweensOf(r, n, nt && nt.vars.overwrite !== !0)._first || Dt(this), this.parent && s !== this.timeline.totalDuration() && kt(this, this._dur * this.timeline._tDur / s, 0, 1), this;
    }
    var a = this._targets, o = r ? X(r) : a, _ = this._ptLookup, h = this._pt, f, l, d, g, c, p, m;
    if ((!n || n === "all") && Or(a, o))
      return n === "all" && (this._pt = 0), Dt(this);
    for (f = this._op = this._op || [], n !== "all" && (I(n) && (c = {}, rt(n, function(y) {
      return c[y] = 1;
    }), n = c), n = ai(a, n)), m = a.length; m--; )
      if (~o.indexOf(a[m])) {
        l = _[m], n === "all" ? (f[m] = n, g = l, d = {}) : (d = f[m] = f[m] || {}, g = n);
        for (c in g)
          p = l && l[c], p && ((!("kill" in p.d) || p.d.kill(c) === !0) && Yt(this, p, "_pt"), delete l[c]), d !== "all" && (d[c] = 1);
      }
    return this._initted && !this._pt && h && Dt(this), this;
  }, t.to = function(r, n) {
    return new t(r, n, arguments[2]);
  }, t.from = function(r, n) {
    return St(1, arguments);
  }, t.delayedCall = function(r, n, s, a) {
    return new t(n, 0, {
      immediateRender: !1,
      lazy: !1,
      overwrite: !1,
      delay: r,
      onComplete: n,
      onReverseComplete: n,
      onCompleteParams: s,
      onReverseCompleteParams: s,
      callbackScope: a
    });
  }, t.fromTo = function(r, n, s) {
    return St(2, arguments);
  }, t.set = function(r, n) {
    return n.duration = 0, n.repeatDelay || (n.repeat = 0), new t(r, n);
  }, t.killTweensOf = function(r, n, s) {
    return A.killTweensOf(r, n, s);
  }, t;
})(Ft);
G(z.prototype, {
  _targets: [],
  _lazy: 0,
  _startAt: 0,
  _op: 0,
  _onInit: 0
});
rt("staggerTo,staggerFrom,staggerFromTo", function(u) {
  z[u] = function() {
    var t = new U(), e = ue.call(arguments, 0);
    return e.splice(u === "staggerFromTo" ? 5 : 4, 0, 0), t[u].apply(t, e);
  };
});
var ke = function(t, e, i) {
  return t[e] = i;
}, fr = function(t, e, i) {
  return t[e](i);
}, oi = function(t, e, i, r) {
  return t[e](r.fp, i);
}, _i = function(t, e, i) {
  return t.setAttribute(e, i);
}, lr = function(t, e) {
  return O(t[e]) ? fr : Fe(t[e]) && t.setAttribute ? _i : ke;
}, dr = function(t, e) {
  return e.set(e.t, e.p, Math.round((e.s + e.c * t) * 1e6) / 1e6, e);
}, hi = function(t, e) {
  return e.set(e.t, e.p, !!(e.s + e.c * t), e);
}, fi = function(t, e) {
  var i = e._pt, r = "";
  if (!t && e.b)
    r = e.b;
  else if (t === 1 && e.e)
    r = e.e;
  else {
    for (; i; )
      r = i.p + (i.m ? i.m(i.s + i.c * t) : Math.round((i.s + i.c * t) * 1e4) / 1e4) + r, i = i._next;
    r += e.c;
  }
  e.set(e.t, e.p, r, e);
}, be = function(t, e) {
  for (var i = e._pt; i; )
    i.r(t, i.d), i = i._next;
}, li = function(t, e, i, r) {
  for (var n = this._pt, s; n; )
    s = n._next, n.p === r && n.modifier(t, e, i), n = s;
}, di = function(t) {
  for (var e = this._pt, i, r; e; )
    r = e._next, e.p === t && !e.op || e.op === t ? Yt(this, e, "_pt") : e.dep || (i = 1), e = r;
  return !i;
}, ci = function(t, e, i, r) {
  r.mSet(t, e, r.m.call(r.tween, i, r.mt), r);
}, pi = function(t) {
  for (var e = t._pt, i, r, n, s; e; ) {
    for (i = e._next, r = n; r && r.pr > e.pr; )
      r = r._next;
    (e._prev = r ? r._prev : s) ? e._prev._next = e : n = e, (e._next = r) ? r._prev = e : s = e, e = i;
  }
  t._pt = n;
}, wt = /* @__PURE__ */ (function() {
  function u(e, i, r, n, s, a, o, _, h) {
    this.t = i, this.s = n, this.c = s, this.p = r, this.r = a || dr, this.d = o || this, this.set = _ || ke, this.pr = h || 0, this._next = e, e && (e._prev = this);
  }
  var t = u.prototype;
  return t.modifier = function(i, r, n) {
    this.mSet = this.mSet || this.set, this.set = ci, this.m = i, this.mt = n, this.tween = r;
  }, u;
})();
rt(ge + "parent,duration,ease,delay,overwrite,runBackwards,startAt,yoyo,immediateRender,repeat,repeatDelay,data,paused,reversed,lazy,callbackScope,stringFilter,id,yoyoEase,stagger,inherit,repeatRefresh,keyframes,autoRevert,scrollTrigger,easeReverse", function(u) {
  return pe[u] = 1;
});
V.TweenMax = V.TweenLite = z;
V.TimelineLite = V.TimelineMax = U;
A = new U({
  sortChildren: !1,
  defaults: At,
  autoRemoveChildren: !0,
  id: "root",
  smoothChildTiming: !0
});
Tt.stringFilter = Xr;
var dt = [], qt = {}, gi = [], Oe = 0, mi = 0, te = function(t) {
  return (qt[t] || gi).map(function(e) {
    return e();
  });
}, fe = function() {
  var t = Date.now(), e = [];
  t - Oe > 2 && (te("matchMediaInit"), dt.forEach(function(i) {
    var r = i.queries, n = i.conditions, s, a, o, _;
    for (a in r)
      s = $.matchMedia(r[a]).matches, s && (o = 1), s !== n[a] && (n[a] = s, _ = 1);
    _ && (i.revert(), o && e.push(i));
  }), te("matchMediaRevert"), e.forEach(function(i) {
    return i.onMatch(i, function(r) {
      return i.add(null, r);
    });
  }), Oe = t, te("matchMedia"));
}, cr = /* @__PURE__ */ (function() {
  function u(e, i) {
    this.selector = i && oe(i), this.data = [], this._r = [], this.isReverted = !1, this.id = mi++, e && this.add(e);
  }
  var t = u.prototype;
  return t.add = function(i, r, n) {
    O(i) && (n = r, r = i, i = O);
    var s = this, a = function() {
      var _ = S, h = s.selector, f;
      return _ && _ !== s && _.data.push(s), n && (s.selector = oe(n)), S = s, f = r.apply(s, arguments), O(f) && s._r.push(f), S = _, s.selector = h, s.isReverted = !1, f;
    };
    return s.last = a, i === O ? a(s, function(o) {
      return s.add(null, o);
    }) : i ? s[i] = a : a;
  }, t.ignore = function(i) {
    var r = S;
    S = null, i(this), S = r;
  }, t.getTweens = function() {
    var i = [];
    return this.data.forEach(function(r) {
      return r instanceof u ? i.push.apply(i, r.getTweens()) : r instanceof z && !(r.parent && r.parent.data === "nested") && i.push(r);
    }), i;
  }, t.clear = function() {
    this._r.length = this.data.length = 0;
  }, t.kill = function(i, r) {
    var n = this;
    if (i ? (function() {
      for (var a = n.getTweens(), o = n.data.length, _; o--; )
        _ = n.data[o], _.data === "isFlip" && (_.revert(), _.getChildren(!0, !0, !1).forEach(function(h) {
          return a.splice(a.indexOf(h), 1);
        }));
      for (a.map(function(h) {
        return {
          g: h._dur || h._delay || h._sat && !h._sat.vars.immediateRender ? h.globalTime(0) : -1 / 0,
          t: h
        };
      }).sort(function(h, f) {
        return f.g - h.g || -1 / 0;
      }).forEach(function(h) {
        return h.t.revert(i);
      }), o = n.data.length; o--; )
        _ = n.data[o], _ instanceof U ? _.data !== "nested" && (_.scrollTrigger && _.scrollTrigger.revert(), _.kill()) : !(_ instanceof z) && _.revert && _.revert(i);
      n._r.forEach(function(h) {
        return h(i, n);
      }), n.isReverted = !0;
    })() : this.data.forEach(function(a) {
      return a.kill && a.kill();
    }), this.clear(), r)
      for (var s = dt.length; s--; )
        dt[s].id === this.id && dt.splice(s, 1);
  }, t.revert = function(i) {
    this.kill(i || {});
  }, u;
})(), yi = /* @__PURE__ */ (function() {
  function u(e) {
    this.contexts = [], this.scope = e, S && S.data.push(this);
  }
  var t = u.prototype;
  return t.add = function(i, r, n) {
    j(i) || (i = {
      matches: i
    });
    var s = new cr(0, n || this.scope), a = s.conditions = {}, o, _, h;
    S && !s.selector && (s.selector = S.selector), this.contexts.push(s), r = s.add("onMatch", r), s.queries = i;
    for (_ in i)
      _ === "all" ? h = 1 : (o = $.matchMedia(i[_]), o && (dt.indexOf(s) < 0 && dt.push(s), (a[_] = o.matches) && (h = 1), o.addListener ? o.addListener(fe) : o.addEventListener("change", fe)));
    return h && r(s, function(f) {
      return s.add(null, f);
    }), this;
  }, t.revert = function(i) {
    this.kill(i || {});
  }, t.kill = function(i) {
    this.contexts.forEach(function(r) {
      return r.kill(i, !0);
    });
  }, u;
})(), Qt = {
  registerPlugin: function() {
    for (var t = arguments.length, e = new Array(t), i = 0; i < t; i++)
      e[i] = arguments[i];
    e.forEach(function(r) {
      return sr(r);
    });
  },
  timeline: function(t) {
    return new U(t);
  },
  getTweensOf: function(t, e) {
    return A.getTweensOf(t, e);
  },
  getProperty: function(t, e, i, r) {
    I(t) && (t = X(t)[0]);
    var n = Pt(t || {}).get, s = i ? Ye : Qe;
    return i === "native" && (i = ""), t && (e ? s((Y[e] && Y[e].get || n)(t, e, i, r)) : function(a, o, _) {
      return s((Y[a] && Y[a].get || n)(t, a, o, _));
    });
  },
  quickSetter: function(t, e, i) {
    if (t = X(t), t.length > 1) {
      var r = t.map(function(h) {
        return ct.quickSetter(h, e, i);
      }), n = r.length;
      return function(h) {
        for (var f = n; f--; )
          r[f](h);
      };
    }
    t = t[0] || {};
    var s = Y[e], a = Pt(t), o = a.harness && (a.harness.aliases || {})[e] || e, _ = s ? function(h) {
      var f = new s();
      mt._pt = 0, f.init(t, i ? h + i : h, mt, 0, [t]), f.render(1, f), mt._pt && be(1, mt);
    } : a.set(t, o);
    return s ? _ : function(h) {
      return _(t, o, i ? h + i : h, a, 1);
    };
  },
  quickTo: function(t, e, i) {
    var r, n = ct.to(t, G((r = {}, r[e] = "+=0.1", r.paused = !0, r.stagger = 0, r), i || {})), s = function(o, _, h) {
      return n.resetTo(e, o, _, h);
    };
    return s.tween = n, s;
  },
  isTweening: function(t) {
    return A.getTweensOf(t, !0).length > 0;
  },
  defaults: function(t) {
    return t && t.ease && (t.ease = lt(t.ease, At.ease)), Pe(At, t || {});
  },
  config: function(t) {
    return Pe(Tt, t || {});
  },
  registerEffect: function(t) {
    var e = t.name, i = t.effect, r = t.plugins, n = t.defaults, s = t.extendTimeline;
    (r || "").split(",").forEach(function(a) {
      return a && !Y[a] && !V[a] && Ot(e + " effect requires " + a + " plugin.");
    }), $t[e] = function(a, o, _) {
      return i(X(a), G(o || {}, n), _);
    }, s && (U.prototype[e] = function(a, o, _) {
      return this.add($t[e](a, j(o) ? o : (_ = o) && {}, this), _);
    });
  },
  registerEase: function(t, e) {
    b[t] = lt(e);
  },
  parseEase: function(t, e) {
    return arguments.length ? lt(t, e) : b;
  },
  getById: function(t) {
    return A.getById(t);
  },
  exportRoot: function(t, e) {
    t === void 0 && (t = {});
    var i = new U(t), r, n;
    for (i.smoothChildTiming = q(t.smoothChildTiming), A.remove(i), i._dp = 0, i._time = i._tTime = A._time, r = A._first; r; )
      n = r._next, (e || !(!r._dur && r instanceof z && r.vars.onComplete === r._targets[0])) && H(i, r, r._start - r._delay), r = n;
    return H(A, i, 0), i;
  },
  context: function(t, e) {
    return t ? new cr(t, e) : S;
  },
  matchMedia: function(t) {
    return new yi(t);
  },
  matchMediaRefresh: function() {
    return dt.forEach(function(t) {
      var e = t.conditions, i, r;
      for (r in e)
        e[r] && (e[r] = !1, i = 1);
      i && t.revert();
    }) || fe();
  },
  addEventListener: function(t, e) {
    var i = qt[t] || (qt[t] = []);
    ~i.indexOf(e) || i.push(e);
  },
  removeEventListener: function(t, e) {
    var i = qt[t], r = i && i.indexOf(e);
    r >= 0 && i.splice(r, 1);
  },
  utils: {
    wrap: Wr,
    wrapYoyo: Qr,
    distribute: Je,
    random: er,
    snap: tr,
    normalize: Gr,
    getUnit: ut,
    clamp: Ur,
    splitColor: ar,
    toArray: X,
    selector: oe,
    mapRange: ir,
    pipe: Br,
    unitize: Vr,
    interpolate: Yr,
    shuffle: je
  },
  install: Ue,
  effects: $t,
  ticker: Z,
  updateRoot: U.updateRoot,
  plugins: Y,
  globalTimeline: A,
  core: {
    PropTween: wt,
    globals: Be,
    Tween: z,
    Timeline: U,
    Animation: Ft,
    getCache: Pt,
    _removeLinkedListItem: Yt,
    reverting: function() {
      return F;
    },
    context: function(t) {
      return t && S && (S.data.push(t), t._ctx = S), S;
    },
    suppressOverwrites: function(t) {
      return le = t;
    }
  }
};
rt("to,from,fromTo,delayedCall,set,killTweensOf", function(u) {
  return Qt[u] = z[u];
});
Z.add(U.updateRoot);
mt = Qt.to({}, {
  duration: 0
});
var Ti = function(t, e) {
  for (var i = t._pt; i && i.p !== e && i.op !== e && i.fp !== e; )
    i = i._next;
  return i;
}, vi = function(t, e) {
  var i = t._targets, r, n, s;
  for (r in e)
    for (n = i.length; n--; )
      s = t._ptLookup[n][r], s && (s = s.d) && (s._pt && (s = Ti(s, r)), s && s.modifier && s.modifier(e[r], t, i[n], r));
}, ee = function(t, e) {
  return {
    name: t,
    headless: 1,
    rawVars: 1,
    //don't pre-process function-based values or "random()" strings.
    init: function(r, n, s) {
      s._onInit = function(a) {
        var o, _;
        if (I(n) && (o = {}, rt(n, function(h) {
          return o[h] = 1;
        }), n = o), e) {
          o = {};
          for (_ in n)
            o[_] = e(n[_]);
          n = o;
        }
        vi(a, n);
      };
    }
  };
}, ct = Qt.registerPlugin({
  name: "attr",
  init: function(t, e, i, r, n) {
    var s, a, o;
    this.tween = i;
    for (s in e)
      o = t.getAttribute(s) || "", a = this.add(t, "setAttribute", (o || 0) + "", e[s], r, n, 0, 0, s), a.op = s, a.b = o, this._props.push(s);
  },
  render: function(t, e) {
    for (var i = e._pt; i; )
      F ? i.set(i.t, i.p, i.b, i) : i.r(t, i.d), i = i._next;
  }
}, {
  name: "endArray",
  headless: 1,
  init: function(t, e) {
    for (var i = e.length; i--; )
      this.add(t, i, t[i] || 0, e[i], 0, 0, 0, 0, 0, 1);
  }
}, ee("roundProps", _e), ee("modifiers"), ee("snap", tr)) || Qt;
z.version = U.version = ct.version = "3.15.0";
Ne = 1;
de() && bt();
b.Power0;
b.Power1;
b.Power2;
b.Power3;
b.Power4;
b.Linear;
b.Quad;
b.Cubic;
b.Quart;
b.Quint;
b.Strong;
b.Elastic;
b.Back;
b.SteppedEase;
b.Bounce;
b.Sine;
b.Expo;
b.Circ;
export {
  Ft as Animation,
  ei as GSCache,
  wt as PropTween,
  U as Timeline,
  U as TimelineLite,
  U as TimelineMax,
  z as Tween,
  z as TweenLite,
  z as TweenMax,
  ni as _checkPlugin,
  yt as _colorExp,
  Xr as _colorStringFilter,
  Tt as _config,
  rt as _forEachName,
  Pt as _getCache,
  Cr as _getProperty,
  lr as _getSetter,
  I as _isString,
  Fe as _isUndefined,
  qe as _missingPlugin,
  kr as _numExp,
  Ie as _numWithUnitExp,
  Ge as _parseRelative,
  Y as _plugins,
  br as _relExp,
  Yt as _removeLinkedListItem,
  fi as _renderComplexString,
  Te as _replaceRandom,
  Sr as _round,
  _e as _roundModifier,
  G as _setDefaults,
  pi as _sortPropTweensByPriority,
  Z as _ticker,
  Ur as clamp,
  ct as default,
  Je as distribute,
  ut as getUnit,
  ct as gsap,
  Yr as interpolate,
  ir as mapRange,
  Gr as normalize,
  Br as pipe,
  er as random,
  oe as selector,
  je as shuffle,
  tr as snap,
  ar as splitColor,
  X as toArray,
  Vr as unitize,
  Wr as wrap,
  Qr as wrapYoyo
};
